
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.2.1, mkdocs-material-7.1.8">
    
    
      
        <title>Gvl launch - Bioinformatics Documentation</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.ca7ac06f.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.f1a3b89f.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font-family:"Roboto";--md-code-font-family:"Roboto Mono"}</style>
      
    
    
    
      <link rel="stylesheet" href="../../extra.css">
    
    
<script async src="https://www.googletagmanager.com/gtag/js?id=G-EMNLJD0DWP"></script>
<script>
    window.dataLayer = window.dataLayer || [];
    function gtag() {
        dataLayer.push(arguments);
    }
    gtag('js', new Date());
    gtag('config', 'G-EMNLJD0DWP');
</script>

    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
  
    
    <script>function __prefix(e){return new URL("../..",location).pathname+"."+e}function __get(e,t=localStorage){return JSON.parse(t.getItem(__prefix(e)))}</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#launching-a-personal-gvl-server-on-the-nectar-research-cloud" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../index.html" title="Bioinformatics Documentation" class="md-header__button md-logo" aria-label="Bioinformatics Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Bioinformatics Documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Gvl launch
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        
<a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    melbournebioinformatics/MelBioInf_docs
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../index.html" title="Bioinformatics Documentation" class="md-nav__button md-logo" aria-label="Bioinformatics Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    Bioinformatics Documentation
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    melbournebioinformatics/MelBioInf_docs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../index.html" class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" type="checkbox" id="__nav_2" >
      
      <label class="md-nav__link" for="__nav_2">
        Tools and skill development
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Tools and skill development" data-md-level="1">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Tools and skill development
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../guides/bioinfo.html" class="md-nav__link">
        What is Bioinformatics?
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../guides/galaxy.html" class="md-nav__link">
        Introduction to the Galaxy Platform
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../galaxy-workflows/galaxy-workflows.html" class="md-nav__link">
        Introduction to Galaxy Workflows
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../python_overview/python_overview.html" class="md-nav__link">
        Introduction to Python
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../using_git/Using_Git.html" class="md-nav__link">
        Introduction to Git and Github
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../hpc/hpc.html" class="md-nav__link">
        Introduction to HPC
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../unix/unix.html" class="md-nav__link">
        Introduction to Unix
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../intro_R_biologists/intro_R_biologists.html" class="md-nav__link">
        Introduction to R for Biologists
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../Genome_browsers/GenomeBrowsers_Intro.html" class="md-nav__link">
        Introduction to Genome Browsers
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_exp_design/rna_seq_experimental_design.html" class="md-nav__link">
        RNA-seq Experimental Design
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../docker/docker.html" class="md-nav__link">
        Containerized Bioinformatics
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../cwl/cwl.html" class="md-nav__link">
        Common Workflow Language for Bioinformatics
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../bionitio/bionitio.html" class="md-nav__link">
        Command-line software development
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" >
      
      <label class="md-nav__link" for="__nav_3">
        Genomics
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Genomics" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Genomics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_1" type="checkbox" id="__nav_3_1" >
      
      <label class="md-nav__link" for="__nav_3_1">
        Variant detection
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Variant detection" data-md-level="2">
        <label class="md-nav__title" for="__nav_3_1">
          <span class="md-nav__icon md-icon"></span>
          Variant detection
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../var_detect_advanced/var_detect_advanced_background.html" class="md-nav__link">
        Introduction to Variant detection
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling_galaxy_1/variant_calling_galaxy_1.html" class="md-nav__link">
        Variant Calling part 1 (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../var_detect_advanced/var_detect_advanced.html" class="md-nav__link">
        Variant Calling part 2 (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../longread_sv_calling/longread_sv_calling.html" class="md-nav__link">
        Long-read Structural Variant Calling
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_2" type="checkbox" id="__nav_3_2" >
      
      <label class="md-nav__link" for="__nav_3_2">
        de novo assembly
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="de novo assembly" data-md-level="2">
        <label class="md-nav__title" for="__nav_3_2">
          <span class="md-nav__icon md-icon"></span>
          de novo assembly
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../assembly/assembly-background.html" class="md-nav__link">
        Introduction to de novo assembly with Velvet
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../assembly/assembly-protocol.html" class="md-nav__link">
        Introduction to de novo genome assembly for Illumina reads
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../assembly/assembly.html" class="md-nav__link">
        de novo assembly of Illumina reads using Velvet (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../assembly/spades.html" class="md-nav__link">
        de novo assembly of Illumina reads using Spades (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../pacbio/index.html" class="md-nav__link">
        de novo assembly of PacBio reads (unix)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../hybrid_assembly/nanopore_assembly.html" class="md-nav__link">
        Hybrid genome assembly - Nanopore and Illumina
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" >
      
      <label class="md-nav__link" for="__nav_4">
        Transcriptomics
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Transcriptomics" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          Transcriptomics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_exp_design/rna_seq_experimental_design.html" class="md-nav__link">
        RNAseq Experimental Design
      </a>
    </li>
  

          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_2" type="checkbox" id="__nav_4_2" >
      
      <label class="md-nav__link" for="__nav_4_2">
        Bulk RNAseq
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Bulk RNAseq" data-md-level="2">
        <label class="md-nav__title" for="__nav_4_2">
          <span class="md-nav__icon md-icon"></span>
          Bulk RNAseq
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_dge_basic/rna_seq_basic_background.html" class="md-nav__link">
        Introduction to bulk RNAseq analysis
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_dge_basic/rna_seq_basic_tutorial.html" class="md-nav__link">
        RNAseq analysis using HISAT2 (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_dge_basic/rna_seq_basic_tuxedo.html" class="md-nav__link">
        RNAseq analysis using Tuxedo (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_dge_advanced/rna_seq_advanced_tutorial.html" class="md-nav__link">
        RNAseq differential expression tool comparision (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_dge_in_r/rna_seq_r.html" class="md-nav__link">
        RNAseq differential expression in R
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_3" type="checkbox" id="__nav_4_3" >
      
      <label class="md-nav__link" for="__nav_4_3">
        Singlecell RNAseq
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Singlecell RNAseq" data-md-level="2">
        <label class="md-nav__title" for="__nav_4_3">
          <span class="md-nav__icon md-icon"></span>
          Singlecell RNAseq
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../singlecell/index.html" class="md-nav__link">
        10X single-cell RNA-seq analysis in R
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" >
      
      <label class="md-nav__link" for="__nav_5">
        Proteomics
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Proteomics" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Proteomics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../proteomics_basic/proteomics_basic.html" class="md-nav__link">
        Identifying proteins from mass spectrometry data
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../proteomics_basic/background_data_formats.html" class="md-nav__link">
        Data Formats Background
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../proteomics_basic/background_protein_databases.html" class="md-nav__link">
        Protein Databases
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../proteomics_basic/background_search_engines.html" class="md-nav__link">
        Search Engines
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../proteomics_basic/background_protein_prophet.html" class="md-nav__link">
        Protein Prophet
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" >
      
      <label class="md-nav__link" for="__nav_6">
        Statistics and Visualisation
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Statistics and Visualisation" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Statistics and Visualisation
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../python_viz/python_viz.html" class="md-nav__link">
        Data visualisation with Python
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_7" type="checkbox" id="__nav_7" >
      
      <label class="md-nav__link" for="__nav_7">
        Structural Modelling
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Structural Modelling" data-md-level="1">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          Structural Modelling
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../molecular_dynamics_101/molecular_dynamics_101.html" class="md-nav__link">
        Molecular Dynamics - Introduction to cluster computing
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../molecular_dynamics_201/molecular_dynamics_201.html" class="md-nav__link">
        Molecular Dynamics - Building input files, visualising the trajectory
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#tutorial-overview" class="md-nav__link">
    Tutorial Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#background" class="md-nav__link">
    Background
  </a>
  
    <nav class="md-nav" aria-label="Background">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#what-is-the-gvl" class="md-nav__link">
    What is the GVL?
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#what-is-nectar" class="md-nav__link">
    What is NeCTAR?
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-1-access-the-nectar-dashboard" class="md-nav__link">
    Section 1: Access the NeCTAR dashboard
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-2-get-your-cloud-credentials" class="md-nav__link">
    Section 2: Get your cloud credentials
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-3-launch-your-personal-gvl-instance" class="md-nav__link">
    Section 3: Launch your personal GVL instance
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-4-access-your-gvl-instance" class="md-nav__link">
    Section 4: Access your GVL instance
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-5-gvl-services" class="md-nav__link">
    Section 5: GVL services
  </a>
  
    <nav class="md-nav" aria-label="Section 5: GVL services">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#galaxy" class="md-nav__link">
    Galaxy
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#cloudman" class="md-nav__link">
    CloudMan
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#lubuntu-desktop" class="md-nav__link">
    Lubuntu Desktop
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ssh" class="md-nav__link">
    SSH
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#jupyterhub" class="md-nav__link">
    JupyterHub
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#rstudio" class="md-nav__link">
    RStudio
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#public-html" class="md-nav__link">
    Public HTML
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#pacbio-smrt-portal" class="md-nav__link">
    PacBio SMRT Portal
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-6-shutting-your-machine-down" class="md-nav__link">
    Section 6: Shutting your machine down
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/melbournebioinformatics/MelBioInf_docs/edit/master/docs/tutorials/gvl_launch/gvl_launch.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                <style type="text/css">
    body{
        line-height: 2;
        font-size: 16px;
    }

    ol li{padding: 4px;}
    ul li{padding: 0px;}
    h4 {margin: 30px 0px 15px 0px;}

    div.code {
        font-family: "Courier New";
        border: 1px solid;
        border-color: #999999;
        background-color: #eeeeee;
        padding: 5px 10px;
        margin: 10px;
        border-radius: 5px;
        overflow: auto;
    }

    div.question {
        color: #666666;
        background-color: #e1eaf9;
        padding: 15px 25px;
        margin: 20px;
        font-size: 15px;
        border-radius: 20px;
    }

    div.question h4 {
        font-style: italic;
        margin: 10px 0px !important;
    }
</style>

<p><img src="../media/gvl_logo.jpg" height=100px align=right></p>
<h1 id="launching-a-personal-gvl-server-on-the-nectar-research-cloud">Launching a Personal GVL Server on the NeCTAR Research Cloud<a class="headerlink" href="#launching-a-personal-gvl-server-on-the-nectar-research-cloud" title="Permanent link">&para;</a></h1>
<hr />
<h2 id="tutorial-overview">Tutorial Overview<a class="headerlink" href="#tutorial-overview" title="Permanent link">&para;</a></h2>
<p>This document guides you to launch your own GVL Analysis platform, with Galaxy,
using your default NeCTAR allocation.</p>
<p>The tutorial will go over:</p>
<ol>
<li>Accessing the NeCTAR dashboard using Australian Access Federation (AAF)
   credentials</li>
<li>Getting your NeCTAR Research Cloud credentials</li>
<li>Launching the GVL image</li>
<li>Accessing your GVL instance</li>
<li>GVL services</li>
<li>Shutting your machine down</li>
</ol>
<hr />
<h2 id="background">Background<a class="headerlink" href="#background" title="Permanent link">&para;</a></h2>
<h4 id="what-is-the-gvl">What is the GVL?<a class="headerlink" href="#what-is-the-gvl" title="Permanent link">&para;</a></h4>
<p>The Genomics Virtual Laboratory (GVL) is a research platform that can be
deployed on the cloud.</p>
<p><img src="../media/gvl_dashboard.png" height=500px style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>A private GVL server is a virtual machine running on the cloud and contains a
pre-installed suite of tools for performing bioinformatics analyses. It differs
from public GVL servers (such as the
<a href="http://galaxy-tut.genome.edu.au/">Galaxy Tutorial Server</a>,
<a href="https://usegalaxy.org.au/">Galaxy Australia</a>, and
<a href="http://galaxy-qld.genome.edu.au/">Galaxy Queensland</a>) by providing full
administrative access to the server, as well as the full suite of GVL services,
whereas public GVL servers provide restricted access for security reasons.
For example, public GVL servers do not provide access to the Ubuntu desktop,
the Linux command line or JupyterHub at present.</p>
<p>Accessing the GVL server is completely free on the Australian NeCTAR Research
Cloud, provided that you have a account with NeCTAR with allocated resources.</p>
<h4 id="what-is-nectar">What is NeCTAR?<a class="headerlink" href="#what-is-nectar" title="Permanent link">&para;</a></h4>
<p>The <a href="http://nectar.org.au/">National eResearch Collaboration Tools and Resources project</a>
(NeCTAR) is an Australian programme that provides computing infrastructure
and services to Australian researchers. The NeCTAR Cloud allows us to
deploy virtual machines as a platform for research.</p>
<p>While it is possible to launch the GVL on <a href="https://aws.amazon.com/">Amazon</a>,
you may have to pay Amazon usage charges (the GVL software itself is free).</p>
<hr />
<h2 id="section-1-access-the-nectar-dashboard">Section 1: Access the NeCTAR dashboard<a class="headerlink" href="#section-1-access-the-nectar-dashboard" title="Permanent link">&para;</a></h2>
<ol>
<li>
<p>Login into the NeCTAR dashboard at
    <a href="https://dashboard.rc.nectar.org.au">dashboard.rc.nectar.org.au</a>.<br />
    Only members of the Australian Access Federation (AAF) can access Australian
    Research Cloud resources. Most Australian universities are members of the
    AAF, however, if you belong to an institution that is not a member of AAF,
    <a href="http://genome.edu.au/help/gvl-help">GVL Help</a> may be able to provide you
    with credentials. You also have the option of using a commercial cloud
    provider such as <a href="https://aws.amazon.com/">AWS</a> to host your server.</p>
</li>
<li>
<p>Choose your organisation from the list and login using your credentials.
    If you are from the University of Melbourne, choose &lsquo;The University of
    Melbourne&rsquo; and not &lsquo;The University of Melbourne (with ECP)&rsquo;.</p>
</li>
<li>
<p>Login with your institutional username and password. If this is the first
    time you have accessed the Australian Research Cloud, you must agree to some
    terms and conditions.</p>
<p>When you log in for the first time, you are automatically allocated a trial
project which lasts for 3 months. This trial project allows you to launch a
medium instance (with 2 cores) which is sufficient for launching a GVL
instance.</p>
<p><img src="../media/nectar_dashboard.png" height="400px" style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>If you have projects which require more compute resources, you can apply
for more allocation <a href="https://dashboard.rc.nectar.org.au/allocation/">here</a>.</p>
</li>
</ol>
<hr />
<h2 id="section-2-get-your-cloud-credentials">Section 2: Get your cloud credentials<a class="headerlink" href="#section-2-get-your-cloud-credentials" title="Permanent link">&para;</a></h2>
<p>Launching a GVL instance requires OpenStack API credentials from NeCTAR. These credentials allows the GVL launcher
to create a new GVL Virtual Machine on your behalf. Obtaining these credentials is a 5 step process.</p>
<p><img src="../media/openstack_creds_annotated_step_1.png" height="400px" /></p>
<ol>
<li>
<p>From the NeCTAR dashboard, on the left sidebar, navigate to<br />
    Project &gt; Compute &gt; Access &amp; Security</p>
</li>
<li>
<p>Click on the top <strong>&lsquo;API Access&rsquo;</strong> tab.</p>
</li>
<li>
<p>Click on the <strong>&lsquo;Download OpenStack RC File&rsquo;</strong> button on the top right.
    A file containing your credentials will be saved to your downloads folder.
    This file will be needed later in the launch process.</p>
</li>
<li>
<p>Next you must obtain your OpenStack password and record it securely for future use.
    If you have ever done this step before, you should reuse your previously saved
    password.</p>
<p>To obtain your OpenStack password, click on your username on the top right hand corner
and go to settings as shown below.</p>
<p><img src="../media/openstack_creds_annotated_step_2.png" /></p>
<p>Click on the reset password link.</p>
<p><img src="../media/openstack_creds_annotated_step_3.png" /></p>
</li>
<li>
<p>Once reset, your password will be displayed. Record this securely for all future
    GVL launches.</p>
<p><img src="../media/openstack_creds_annotated_step_4.png" /></p>
</li>
</ol>
<hr />
<h2 id="section-3-launch-your-personal-gvl-instance">Section 3: Launch your personal GVL instance<a class="headerlink" href="#section-3-launch-your-personal-gvl-instance" title="Permanent link">&para;</a></h2>
<ol>
<li>
<p>In a new browser tab, go to <a href="https://launch.usegalaxy.org">launch.usegalaxy.org</a></p>
</li>
<li>
<p>You will see the screen below. Select the first option from the list - &ldquo;Genomics Virtual Lab&rdquo;.</p>
<p><img src="../media/cloudlaunch_gvl_step_1.png" height="400px" /> </p>
</li>
<li>
<p>You will be asked to login with your preferred social network account.</p>
<p><img src="../media/cloudlaunch_gvl_step_2.png" /> </p>
</li>
<li>
<p>Once logged in, perform the following steps.</p>
<p><img src="../media/cloudlaunch_gvl_step_3.png" style="display:block; margin-left: auto; margin-right:auto; margin-top:10px;"></p>
<ul>
<li>
<ol>
<li>Select NeCTAR for the question &ldquo;On which cloud would you like to launch your appliance&rdquo;</li>
</ol>
</li>
<li>
<ol>
<li>Click &ldquo;load credentials from file&rdquo; and provide file you downloaded in Section 2.3</li>
</ol>
</li>
<li>
<ol>
<li>Provide the OpenStack password you obtained in Section 2.5.</li>
</ol>
</li>
<li>
<ol>
<li>Click &ldquo;Test and Use these Credentials&rdquo;. The Next button will now be activated.</li>
</ol>
</li>
</ul>
</li>
<li>
<p>Click the next button, and provide the following options.</p>
<p><img src="../media/cloudlaunch_gvl_step_4.png" /> </p>
<ul>
<li><strong>Password:</strong> Choose a strong password and remember it. This is the
  password you will use later to log into your instance.</li>
</ul>
</li>
<li>
<p>Optional advanced options<br />
    Toggle the &lsquo;Advanced cloudlaunch options&rsquo; option to see more options. For
    this tutorial, it is not necessary to modify any of the advanced options.</p>
<ul>
<li><strong>Deployment name:</strong> You can override the name with a name of your choice.
  It is recommended you choose a unique name if you launch multiple instances.</li>
<li><strong>Instance type:</strong> Keep the default: Medium (2 vcpu / 8GB RAM)</li>
<li><strong>Root Volume Storage:</strong> Keep the default: Instance storage</li>
<li><strong>Placement Zone:</strong> You can also choose the region of where your server is
  hosted. If you are doing lots of data transfer, it may be beneficial to
  pick a location close to your physical location. More information about
  zones can be found <a href="https://support.rc.nectar.org.au/docs/availability-zones">here</a>.</li>
<li><strong>Key pair:</strong> Key pairs are SSH credentials that can be used to access
  your instance. You can create and import key pairs in the NeCTAR dashboard
  by navigating to Project &gt; Compute &gt; Access &amp; Security &gt; Key Pairs and
  creating or importing a key pair.</li>
</ul>
</li>
<li>
<p>Click <strong>&lsquo;Launch&rsquo;</strong> to launch a GVL.<br />
    The launch process takes 2-5 minutes to start the machine and another 5
    minutes to start and configure Galaxy.</p>
<p><img src="../media/cloudlaunch_gvl_wait.png" style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>If an error occurs, or the launch does not complete in 10 minutes, navigate back to the launch page and try
selecting a different availability zone under the advanced cloudlaunch options -&gt; <strong>Placement Zone</strong> field.</p>
</li>
</ol>
<hr />
<h2 id="section-4-access-your-gvl-instance">Section 4: Access your GVL instance<a class="headerlink" href="#section-4-access-your-gvl-instance" title="Permanent link">&para;</a></h2>
<ol>
<li>
<p>Once your instance has finished launching, click on the Access address
    to access your GVL dashboard.</p>
<p><img src="../media/cloudlaunch_gvl_ready.png" style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>If you accidentally closed the launch page, you can find your access address at any time
by logging back into <a href="https://launch.usegalaxy.org">launch.usegalaxy.org</a> and navigating
to the &ldquo;My Appliances&rdquo; section through the menu bar.</p>
</li>
<li>
<p>Explore the GVL dashboard.
    Have a read through of the services provided by the GVL.</p>
</li>
</ol>
<p><img src="../media/gvl_dashboard.png" height=500px style="display:block; margin-left: auto; margin-right:auto;"></p>
<hr />
<h2 id="section-5-gvl-services">Section 5: GVL services<a class="headerlink" href="#section-5-gvl-services" title="Permanent link">&para;</a></h2>
<p>Listed below are short descriptions of the services the GVL provides.</p>
<h4 id="galaxy">Galaxy<a class="headerlink" href="#galaxy" title="Permanent link">&para;</a></h4>
<p><a href="https://galaxyproject.org/">Galaxy</a> is a web-based platform for computational
biomedical research. The GVL has a number of Galaxy tutorials available
<a href="http://genome.edu.au/learn">here</a>.</p>
<p><img src="../media/gvl_galaxy.png" height=400px style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>To begin using Galaxy, register as a user by navigating to User &gt; Register
on the top Galaxy bar.</p>
<h4 id="cloudman">CloudMan<a class="headerlink" href="#cloudman" title="Permanent link">&para;</a></h4>
<p>CloudMan is a cloud manager that manages services on your GVL instance. Use
Cloudman to start and manage your Galaxy service and to add additional nodes
to your compute cluster (if you have enough resources).</p>
<p>You can log into CloudMan by using the username &lsquo;ubuntu&rsquo; and your cluster
password.</p>
<p><img src="../media/cloudman.png" height=350px style="display:block; margin-left: auto; margin-right:auto;"></p>
<p>You can also shut down your instance (permanently) with CloudMan.</p>
<h4 id="lubuntu-desktop">Lubuntu Desktop<a class="headerlink" href="#lubuntu-desktop" title="Permanent link">&para;</a></h4>
<p>Lubuntu is a lightweight desktop environment through which you can run desktop
applications on your virtual machine through your web browser. You can also
access the GVL command line utilities through the desktop.</p>
<p>You can log into Lubuntu Desktop using the username &lsquo;ubuntu&rsquo; and your cluster
password.</p>
<p><img src="../media/lubuntu.png" height=350px style="display:block; margin-left: auto; margin-right:auto;"></p>
<h4 id="ssh">SSH<a class="headerlink" href="#ssh" title="Permanent link">&para;</a></h4>
<p>Secure Shell (SSH) is a network protocol that allows us to connect to a remotely
machine. You can login to your virtual machine remotely through an SSH client.</p>
<p>If you are using Windows, you will need to download an SSH client such as
<a href="http://www.chiark.greenend.org.uk/~sgtatham/putty/">PuTTY</a>. If you are using
OSX, open up a Terminal window.</p>
<p>If you are unfamiliar with the command line and UNIX, many
<a href="http://swcarpentry.github.io/shell-novice/">tutorials</a>
<a href="../unix/">on UNIX</a>
<a href="http://www.ee.surrey.ac.uk/Teaching/Unix/">can be</a>
<a href="http://www.doc.ic.ac.uk/~wjk/UnixIntro/">found online</a>.</p>
<p>You can ssh into your machine using the either the username &lsquo;ubuntu&rsquo; or the
username &lsquo;researcher&rsquo; and using your cluster password. It is recommended to use
the researcher account when you are doing your computational research and use
the ubuntu account when you need administrative powers (such as installing
software).</p>
<p><img src="../media/command_line.png" height=350px style="display:block; margin-left: auto; margin-right:auto;"></p>
<h4 id="jupyterhub">JupyterHub<a class="headerlink" href="#jupyterhub" title="Permanent link">&para;</a></h4>
<p><a href="http://jupyter.org/">JupyterHub</a> is a web-based interactive computational
environment where you can combine code execution, text, mathematics, plots and
rich media into a single document. Currently, JupyterHub can connect to Python2
and Python3 kernals.</p>
<p>If you are unfamiliar with Python, there are many
<a href="http://swcarpentry.github.io/python-novice-inflammation/">tutorials</a>
<a href="../python_overview/python_overview/">available</a>
<a href="http://pythonprogramminglanguage.com/">online</a>.</p>
<p>You can log into JupyterHub with the username &lsquo;researcher&rsquo; and your cluster
password.</p>
<p>You may need to install Python packages you intend to use via the command line
beforehand.</p>
<p><img src="../media/jupyterhub.png" height=350px style="display:block; margin-left: auto; margin-right:auto;"></p>
<h4 id="rstudio">RStudio<a class="headerlink" href="#rstudio" title="Permanent link">&para;</a></h4>
<p>RStudio Server gives browser-based access to RStudio, the popular programming
and analysis environment for the R programming language. You can find out more
about RStudio <a href="https://www.rstudio.com/home/">here</a>, and the R programming
language <a href="https://www.r-project.org/">here</a>.</p>
<p>You can log into RStudio with the username &lsquo;researcher&rsquo; and your cluster
password.</p>
<p><img src="../media/rstudio.png" height=350px style="display:block; margin-left: auto; margin-right:auto;"></p>
<h4 id="public-html">Public HTML<a class="headerlink" href="#public-html" title="Permanent link">&para;</a></h4>
<p>This is a shared web-accessible folder. Any files you place in the directory
<code>/home/researcher/public_html</code> will be publicly accessible.</p>
<h4 id="pacbio-smrt-portal">PacBio SMRT Portal<a class="headerlink" href="#pacbio-smrt-portal" title="Permanent link">&para;</a></h4>
<p><a href="http://www.pacb.com/products-and-services/analytical-software/smrt-analysis/">PacBio&rsquo;s SMRT Portal</a> is an open source software suite for the analysis of single molecule, real-time sequencing data.</p>
<p>Before you use SMRT Portal, you need to firstly install it through the Admin console. <em>Please note PacBio recommends the use of a 16 core instance with 64GB of RAM (or higher) for this package.</em></p>
<p>To install SMRT Portal from the GVL dashboard:</p>
<ul>
<li>
<p>Click on &lsquo;Admin&rsquo; in the top navigation bar.</p>
</li>
<li>
<p>Log in with the username &lsquo;ubuntu&rsquo; and your cluster password. The screen will now show all the tools available to be installed.</p>
</li>
<li>
<p>Scroll down to &lsquo;SMRT Analysis&rsquo;, and click &lsquo;install&rsquo;.</p>
</li>
<li>
<p>The install is complete when SMRT Portal is available as a tool on the GVL dashboard, and a green tick is displayed.</p>
</li>
</ul>
<p>When launching SMRT Portal for the first time, you will need to register yourself as a new user.</p>
<hr />
<h2 id="section-6-shutting-your-machine-down">Section 6: Shutting your machine down<a class="headerlink" href="#section-6-shutting-your-machine-down" title="Permanent link">&para;</a></h2>
<p>There are two ways to terminate your instance. Terminating your instance is
permanent and all data will be deleted (unless you have persistent
<a href="https://support.rc.nectar.org.au/docs/volumes">volume storage</a> which you will
need to apply for).</p>
<ol>
<li>
<p>Via Cloudman<br />
    Log into CloudMan by using the username &lsquo;ubuntu&rsquo; and your cluster password.
    Click the <strong>Shut down&hellip;</strong> button under Cluster Controls.</p>
</li>
<li>
<p>Via the NeCTAR dashboard<br />
    Navigate to the <a href="https://dashboard.rc.nectar.org.au/project/instances/">Instances</a>
    page by navigating to Project &gt; Compute &gt; Instances on the left panel. Find
    the instance you want to terminate, and on the right-most column (Actions),
    click on the arrow button, and select <strong>Terminate Instance</strong>.</p>
</li>
</ol>
                
              
              
                


              
            </article>
          </div>
        </div>
        
      </main>
      
        
<footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            <a href="https://www.melbournebioinformatics.org.au/">Melbourne Bioinformatics</a>
          </div>
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
        
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../..", "features": [], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../../assets/javascripts/workers/search.b0710199.min.js", "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.76f349be.min.js"></script>
      
    
  </body>
</html>