
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.2.1, mkdocs-material-7.1.8">
    
    
      
        <title>Galaxy 101 - Bioinformatics Documentation</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.ca7ac06f.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.f1a3b89f.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font-family:"Roboto";--md-code-font-family:"Roboto Mono"}</style>
      
    
    
    
      <link rel="stylesheet" href="../../extra.css">
    
    
<script async src="https://www.googletagmanager.com/gtag/js?id=G-EMNLJD0DWP"></script>
<script>
    window.dataLayer = window.dataLayer || [];
    function gtag() {
        dataLayer.push(arguments);
    }
    gtag('js', new Date());
    gtag('config', 'G-EMNLJD0DWP');
</script>

    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
  
    
    <script>function __prefix(e){return new URL("../..",location).pathname+"."+e}function __get(e,t=localStorage){return JSON.parse(t.getItem(__prefix(e)))}</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#introduction-to-galaxy" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../index.html" title="Bioinformatics Documentation" class="md-header__button md-logo" aria-label="Bioinformatics Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Bioinformatics Documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Galaxy 101
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        
<a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    melbournebioinformatics/MelBioInf_docs
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../index.html" title="Bioinformatics Documentation" class="md-nav__button md-logo" aria-label="Bioinformatics Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    Bioinformatics Documentation
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    melbournebioinformatics/MelBioInf_docs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../index.html" class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" type="checkbox" id="__nav_2" >
      
      <label class="md-nav__link" for="__nav_2">
        Tools and skill development
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Tools and skill development" data-md-level="1">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Tools and skill development
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../guides/bioinfo.html" class="md-nav__link">
        What is Bioinformatics?
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../guides/galaxy.html" class="md-nav__link">
        Introduction to the Galaxy Platform
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../galaxy-workflows/galaxy-workflows.html" class="md-nav__link">
        Introduction to Galaxy Workflows
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../python_overview/python_overview.html" class="md-nav__link">
        Introduction to Python
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../using_git/Using_Git.html" class="md-nav__link">
        Introduction to Git and Github
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../hpc/hpc.html" class="md-nav__link">
        Introduction to HPC
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../unix/unix.html" class="md-nav__link">
        Introduction to Unix
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../intro_R_biologists/intro_R_biologists.html" class="md-nav__link">
        Introduction to R for Biologists
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../Genome_browsers/GenomeBrowsers_Intro.html" class="md-nav__link">
        Introduction to Genome Browsers
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_exp_design/rna_seq_experimental_design.html" class="md-nav__link">
        RNA-seq Experimental Design
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../docker/docker.html" class="md-nav__link">
        Containerized Bioinformatics
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../cwl/cwl.html" class="md-nav__link">
        Common Workflow Language for Bioinformatics
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../bionitio/bionitio.html" class="md-nav__link">
        Command-line software development
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" >
      
      <label class="md-nav__link" for="__nav_3">
        Genomics
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Genomics" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Genomics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_1" type="checkbox" id="__nav_3_1" >
      
      <label class="md-nav__link" for="__nav_3_1">
        Variant detection
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Variant detection" data-md-level="2">
        <label class="md-nav__title" for="__nav_3_1">
          <span class="md-nav__icon md-icon"></span>
          Variant detection
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../var_detect_advanced/var_detect_advanced_background.html" class="md-nav__link">
        Introduction to Variant detection
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling_galaxy_1/variant_calling_galaxy_1.html" class="md-nav__link">
        Variant Calling part 1 (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../var_detect_advanced/var_detect_advanced.html" class="md-nav__link">
        Variant Calling part 2 (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../longread_sv_calling/longread_sv_calling.html" class="md-nav__link">
        Long-read Structural Variant Calling
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_2" type="checkbox" id="__nav_3_2" >
      
      <label class="md-nav__link" for="__nav_3_2">
        de novo assembly
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="de novo assembly" data-md-level="2">
        <label class="md-nav__title" for="__nav_3_2">
          <span class="md-nav__icon md-icon"></span>
          de novo assembly
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../assembly/assembly-background.html" class="md-nav__link">
        Introduction to de novo assembly with Velvet
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../assembly/assembly-protocol.html" class="md-nav__link">
        Introduction to de novo genome assembly for Illumina reads
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../assembly/assembly.html" class="md-nav__link">
        de novo assembly of Illumina reads using Velvet (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../assembly/spades.html" class="md-nav__link">
        de novo assembly of Illumina reads using Spades (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../pacbio/index.html" class="md-nav__link">
        de novo assembly of PacBio reads (unix)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../hybrid_assembly/nanopore_assembly.html" class="md-nav__link">
        Hybrid genome assembly - Nanopore and Illumina
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" >
      
      <label class="md-nav__link" for="__nav_4">
        Transcriptomics
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Transcriptomics" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          Transcriptomics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_exp_design/rna_seq_experimental_design.html" class="md-nav__link">
        RNAseq Experimental Design
      </a>
    </li>
  

          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_2" type="checkbox" id="__nav_4_2" >
      
      <label class="md-nav__link" for="__nav_4_2">
        Bulk RNAseq
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Bulk RNAseq" data-md-level="2">
        <label class="md-nav__title" for="__nav_4_2">
          <span class="md-nav__icon md-icon"></span>
          Bulk RNAseq
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_dge_basic/rna_seq_basic_background.html" class="md-nav__link">
        Introduction to bulk RNAseq analysis
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_dge_basic/rna_seq_basic_tutorial.html" class="md-nav__link">
        RNAseq analysis using HISAT2 (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_dge_basic/rna_seq_basic_tuxedo.html" class="md-nav__link">
        RNAseq analysis using Tuxedo (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_dge_advanced/rna_seq_advanced_tutorial.html" class="md-nav__link">
        RNAseq differential expression tool comparision (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_dge_in_r/rna_seq_r.html" class="md-nav__link">
        RNAseq differential expression in R
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_3" type="checkbox" id="__nav_4_3" >
      
      <label class="md-nav__link" for="__nav_4_3">
        Singlecell RNAseq
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Singlecell RNAseq" data-md-level="2">
        <label class="md-nav__title" for="__nav_4_3">
          <span class="md-nav__icon md-icon"></span>
          Singlecell RNAseq
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../singlecell/index.html" class="md-nav__link">
        10X single-cell RNA-seq analysis in R
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" >
      
      <label class="md-nav__link" for="__nav_5">
        Proteomics
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Proteomics" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Proteomics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../proteomics_basic/proteomics_basic.html" class="md-nav__link">
        Identifying proteins from mass spectrometry data
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../proteomics_basic/background_data_formats.html" class="md-nav__link">
        Data Formats Background
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../proteomics_basic/background_protein_databases.html" class="md-nav__link">
        Protein Databases
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../proteomics_basic/background_search_engines.html" class="md-nav__link">
        Search Engines
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../proteomics_basic/background_protein_prophet.html" class="md-nav__link">
        Protein Prophet
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" >
      
      <label class="md-nav__link" for="__nav_6">
        Statistics and Visualisation
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Statistics and Visualisation" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Statistics and Visualisation
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../python_viz/python_viz.html" class="md-nav__link">
        Data visualisation with Python
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_7" type="checkbox" id="__nav_7" >
      
      <label class="md-nav__link" for="__nav_7">
        Structural Modelling
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Structural Modelling" data-md-level="1">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          Structural Modelling
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../molecular_dynamics_101/molecular_dynamics_101.html" class="md-nav__link">
        Molecular Dynamics - Introduction to cluster computing
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../molecular_dynamics_201/molecular_dynamics_201.html" class="md-nav__link">
        Molecular Dynamics - Building input files, visualising the trajectory
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" class="md-nav__link">
    Learning Objectives
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#requirements" class="md-nav__link">
    Requirements
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#background" class="md-nav__link">
    Background
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-1-preparation" class="md-nav__link">
    Section 1: Preparation.
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-2-getting-data-into-galaxy" class="md-nav__link">
    Section 2: Getting data into Galaxy
  </a>
  
    <nav class="md-nav" aria-label="Section 2: Getting data into Galaxy">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#method-1-upload-a-file-from-your-own-computer" class="md-nav__link">
    Method 1: Upload a file from your own computer
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#method-2-upload-a-file-from-a-url" class="md-nav__link">
    Method 2: Upload a file from a URL
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#method-2-again-get-data-from-a-data-library" class="md-nav__link">
    Method 2 (again): Get data from a Data Library
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-data" class="md-nav__link">
    The data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-3-play-with-the-tools" class="md-nav__link">
    Section 3: Play with the tools
  </a>
  
    <nav class="md-nav" aria-label="Section 3: Play with the tools">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#example-1-histogram-and-summary-statistics" class="md-nav__link">
    Example 1: Histogram and summary statistics
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#example-2-convert-fastq-to-fasta" class="md-nav__link">
    Example 2: Convert Fastq to Fasta
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#example-3-find-ribosomal-rna-features-in-a-dna-sequence" class="md-nav__link">
    Example 3: Find Ribosomal RNA Features in a DNA Sequence
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#what-now" class="md-nav__link">
    What now?
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/melbournebioinformatics/MelBioInf_docs/edit/master/docs/tutorials/galaxy_101/galaxy_101.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                <style>
  .image-header-gvl {
    -webkit-column-count: 3;
    -moz-column-count: 3;
    column-count: 3;

    -webkit-column-gap: 140px;
    -moz-column-gap: 140px;
    column-gap: 140px;
  }
</style>
<div class="image-header-gvl">
    <a href=http://genome.edu.au><img src="../media/gvl_logo.jpg" alt="GVL logo" width="112"/></a>
    <a href="http://galaxyproject.org"><img src="../media/GalaxyLogoHighRes.png" width="164" /></a>
    <a href="https://www.melbournebioinformatics.org.au/"><img src="../../../img/melbioinf_logo.png" alt="Melbourne Bioinformatics logo" width="164"/></a>
</div>

<h1 id="introduction-to-galaxy">Introduction to Galaxy<a class="headerlink" href="#introduction-to-galaxy" title="Permanent link">&para;</a></h1>
<p>Written and maintained by <a href="mailto:simon.gladman@unimelb.edu.au">Simon Gladman</a> - Melbourne Bioinformatics (formerly VLSCI)</p>
<h2 id="overview">Overview<a class="headerlink" href="#overview" title="Permanent link">&para;</a></h2>
<p>This beginners tutorial will introduce Galaxy&rsquo;s interface, tool use, histories, and get new users of the Genomics Virtual Laboratory up and running. You can follow this tutorial with the <a href="../galaxy-workflows/galaxy-workflows.html">Galaxy Workflows</a> tutorial to learn about workflows.</p>
<p>Galaxy is an open source, web-based platform for accessible, reproducible, and transparent computational biomedical research. It allows users without programming experience to easily specify parameters and run individual tools as well as larger workflows. It also captures run information so that any user can repeat and understand a complete computational analysis. Finally, it allows users to share and publish analyses via the web.</p>
<h2 id="learning-objectives">Learning Objectives<a class="headerlink" href="#learning-objectives" title="Permanent link">&para;</a></h2>
<p>At the end of the course, you will be able to:</p>
<ul>
<li>login to a Galaxy server.</li>
<li>upload data to a Galaxy server from:</li>
<li>A file on your local computer</li>
<li>A file on a remote datastore with an accessible URL.</li>
<li>use tools in Galaxy by:</li>
<li>Accessing the tool via the tool menu</li>
<li>Using the tool interface to run the particular tool</li>
<li>Viewing/accessing the tool output.</li>
</ul>
<h2 id="requirements">Requirements<a class="headerlink" href="#requirements" title="Permanent link">&para;</a></h2>
<p>This is a hands-on workshop and attendees should bring their own laptops.</p>
<h2 id="background">Background<a class="headerlink" href="#background" title="Permanent link">&para;</a></h2>
<p>Galaxy is a web based analysis and workflow platform designed for biologists to analyse their own data. It comes with most of the popular bioinformatics tools already installed and ready for use. There are many Galaxy servers around the world and some are tailored with specific toolsets and reference data for analysis of human genomics, microbial genomics, proteomics etc.</p>
<p>There are some introductory slides available <a href="https://docs.google.com/presentation/d/1dzHagGkswjH7MOZ7OACVXGU-riBs33K3J5lWpnCpPhs/pub?start=false&amp;loop=false&amp;slide=id.g537781ff1_2_18">here</a>.</p>
<p>Basically, the Galaxy interface is separated into 3 parts. The tool list on the left, the viewing pane in the middle and the analysis and data history on the right. We will be looking at all 3 parts in this tutorial.</p>
<p><img src="../media/Galaxy_components.png" alt="Galaxy interface components diagram" width="70%" /></p>
<p>This workshop/tutorial will familiarize you with the Galaxy interface. It will cover the following topics:</p>
<ul>
<li>Logging in to the server</li>
<li>Getting data into galaxy</li>
<li>How to access the tools</li>
<li>Using to use some common tools</li>
</ul>
<hr />
<h2 id="section-1-preparation">Section 1: Preparation.<a class="headerlink" href="#section-1-preparation" title="Permanent link">&para;</a></h2>
<p>The purpose of this section is to get you to log in to the server.</p>
<ol>
<li>
<p>Open your browser. We recommend Firefox or Chrome (please don&rsquo;t use Internet Explorer or Safari).</p>
<ul>
<li>Go to the <a href="https://usegalaxy.org.au/">Galaxy Australia</a> server.</li>
<li>Alternatively, you can use a different Galaxy server - a list of available servers is <a href="https://galaxyproject.org/galaxy-services/">here</a>.</li>
</ul>
</li>
<li>
<p>If you have previously registered on this server just log in:</p>
<ul>
<li>On the top menu select: <strong>User -&gt; Login</strong></li>
<li>Enter your password</li>
<li>Click <strong>Submit</strong></li>
</ul>
</li>
<li>
<p>If you haven’t registered on this server, you’ll need to now.</p>
<ul>
<li>On the top menu select: <strong>User -&gt; Register</strong></li>
<li>Enter your email, choose a password, repeat it and add a (all lower case) one word name</li>
<li>Click <strong>Submit</strong></li>
</ul>
</li>
</ol>
<hr />
<h2 id="section-2-getting-data-into-galaxy">Section 2: Getting data into Galaxy<a class="headerlink" href="#section-2-getting-data-into-galaxy" title="Permanent link">&para;</a></h2>
<p>There are 2 main ways to get your data into Galaxy. We will use each of these methods for 3 files and then use those 3 files for the rest of the workshop.</p>
<ul>
<li>
<p>Start a new history for this workshop. To do this:</p>
<ul>
<li>Click on the history menu button (the <img src="../media/Galaxy-menu.png" width=20 /> icon) at the top of the Histories panel.</li>
<li>Select <strong>Create New</strong></li>
</ul>
</li>
</ul>
<p>It is important to note that Galaxy has the concept of &ldquo;File Type&rdquo; built in. This means that each file stored needs to have its type described to Galaxy as it is being made available. Examples of file types are: text, fasta, fastq, vcf, GFF, Genbank, tabular etc.</p>
<p>We will tell Galaxy what type of file each one is as we upload it.</p>
<h3 id="method-1-upload-a-file-from-your-own-computer">Method 1: Upload a file from your own computer<a class="headerlink" href="#method-1-upload-a-file-from-your-own-computer" title="Permanent link">&para;</a></h3>
<p>With this method you can get most of the files on your own computer into Galaxy. (there is a size limit)</p>
<ul>
<li>
<p>Download the following file to your computer: <em><a href="https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/galaxy101/Contig_stats.txt.gz">https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/galaxy101/Contig_stats.txt.gz</a></em></p>
<ul>
<li>(To download this file, copy the link into a new browser tab, and press enter. The file should now download.)</li>
<li>From the Galaxy tool panel, click on <strong>Get Data -&gt; Upload File</strong></li>
<li>Click the <strong>Choose File</strong> button</li>
<li>Find and select the <em>Contig_stats.txt.gz</em> file you downloaded and click <strong>Open</strong></li>
<li>Set the &ldquo;Type&rdquo; (= file format) to <em>tabular</em></li>
<li>Click the <strong>Start</strong> button</li>
<li>Once the progress bar reaches 100%, click the <strong>Close</strong> button</li>
</ul>
</li>
</ul>
<p>The file will now upload to your current history.</p>
<h3 id="method-2-upload-a-file-from-a-url">Method 2: Upload a file from a URL<a class="headerlink" href="#method-2-upload-a-file-from-a-url" title="Permanent link">&para;</a></h3>
<p>If a file exists on a web resource somewhere and you know its URL (Unique resource location - a web address) you can directly load it into Galaxy.</p>
<ul>
<li>
<p>From the tool panel, click on <strong>Get Data -&gt; Upload File</strong></p>
<ul>
<li>Click on the <strong>Paste/Fetch Data</strong> button</li>
<li>Copy and paste the following web address into the URL/Text box: <em><a href="https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/COMP90014/Assignment1/bacterial_std_err_1.fastq.gz">https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/COMP90014/Assignment1/bacterial_std_err_1.fastq.gz</a></em></li>
<li>Set the file format to <em>fastqsanger</em> (not fastqcsanger)</li>
<li>Click <strong>Start</strong></li>
<li>Once the progress bar has reached 100%, click <strong>Close</strong></li>
</ul>
</li>
</ul>
<p>Note that Galaxy is smart enough to recognize that this is a compressed file and so it will uncompress it as it loads it.</p>
<h3 id="method-2-again-get-data-from-a-data-library">Method 2 (again): Get data from a Data Library<a class="headerlink" href="#method-2-again-get-data-from-a-data-library" title="Permanent link">&para;</a></h3>
<p>Now we are going to get another file from a shared Data Library.</p>
<ul>
<li>
<p>Go to the menu at the top of the screen and click <strong>Shared Data -&gt; Data Libraries</strong>.</p>
</li>
<li>
<p>Click on the Library: &ldquo;Galaxy Australia Training Material&rdquo; then &ldquo;Galaxy_101&rdquo;</p>
</li>
<li>
<p>To add the MRSA0252.fa file to our history</p>
</li>
<li>click on the checkbox next to it</li>
<li>Then click the &ldquo;To History&rdquo; button at the top of the page and select &ldquo;As Datasets&rdquo;</li>
<li>Click the &ldquo;Import&rdquo; button</li>
<li>Finally, click &ldquo;Analyse Data&rdquo; in the menu at the top of the screen to return to your history.</li>
</ul>
<p>The DNA sequence of <em>Staphlococcus aureus MRSA252</em> will be loaded into your history as a fasta file.</p>
<p>Your history should now look like this.</p>
<p><img src="../media/starting-history.png" width="300px" align="center" /></p>
<h3 id="the-data">The data<a class="headerlink" href="#the-data" title="Permanent link">&para;</a></h3>
<p>Though we aren&rsquo;t going to focus on the contents of these files and what they mean from a bioinformatics standpoint, here is a brief description of each one.</p>
<ul>
<li>
<p><em>Contigs_stats.txt</em></p>
<ul>
<li>this file contains a table of summary data from a de novo genome assembly (the process of attempting to recover the full genome of an organism from the short read sequences produced by most DNA sequencing machines. )</li>
<li>The columns contain a lot of information but the ones we will be using indicate the amount of data (or coverage) that went into making up each piece of the final assembly.  </li>
</ul>
</li>
</ul>
<!-- -->

<ul>
<li>
<p><em>bacterial_std_err_1.fastq.gz</em></p>
<ul>
<li>This file contains sequence reads as they would come off an Illumina sequencing machine. They are in <a href="https://en.wikipedia.org/wiki/FASTQ_format">fastq</a> format.  </li>
</ul>
</li>
</ul>
<!-- -->

<ul>
<li>
<p><em>MRSA0252.fna</em></p>
<ul>
<li>This file contains the genome sequence of <em>Staphylococcus aureus MRSA252</em>. It is in <a href="https://en.wikipedia.org/wiki/FASTA_format">fasta</a> format.</li>
</ul>
</li>
</ul>
<hr />
<h2 id="section-3-play-with-the-tools">Section 3: Play with the tools<a class="headerlink" href="#section-3-play-with-the-tools" title="Permanent link">&para;</a></h2>
<p>The purpose of this section is to get you used to using the available tools in Galaxy and point out some of the more basic manipulation tools.</p>
<p>Firstly however, you’ll notice that two of the files have very long and confusing names. So we might want to change them. To do this we need to “edit” the file. So:</p>
<ol>
<li>Click on the <img src="../media/Galaxy-edit.png" width=20 /> icon (edit) next to the file in the history called: <em><a href="https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/COMP90014/Assignment1/bacterial_std_err_1.fastq">https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/COMP90014/Assignment1/bacterial_std_err_1.fastq</a></em></li>
<li>In the &ldquo;Name&rdquo; text box, give it a new name. Call it: <em>Typical Fastq File</em></li>
<li>Click the <strong>Save</strong> button.</li>
</ol>
<p>Repeat the process for the MRSA252 fasta file. Rename it to <em>MRSA252.fna</em></p>
<p>Now that’s better. There was a lot of other functionality hidden behind that edit (<img src="../media/Galaxy-edit.png" width=20 />) icon. You can change a file’s data type, convert its format and many other things. Feel free to play around with them at a later date.</p>
<p>Ok, back to the tools.</p>
<h3 id="example-1-histogram-and-summary-statistics">Example 1: Histogram and summary statistics<a class="headerlink" href="#example-1-histogram-and-summary-statistics" title="Permanent link">&para;</a></h3>
<p>The first thing we are going to do is produce a histogram of contig read coverage depths and calculate the summary statistics from the Contig_stats.txt file. To do this we need to cut out a couple of columns, remove a line and then produce a histogram. This will introduce some of the text manipulation tools.</p>
<p>Click on the <img src="../media/Galaxy-view.png" width=20 /> icon of the <em>Contig_stats.txt</em> file to have a look at it. Note that there are 18 columns in this file. We want column 1 and column 6. To do this:</p>
<p><strong>1. Cut out column 1 and column 6.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Text Manipulation -&gt; Cut</strong> and set the following:</li>
<li>Set &ldquo;Cut Columns&rdquo; to: <em>c1,c6</em></li>
<li>&ldquo;Delimited by&rdquo;: <em>Tab</em></li>
<li>&ldquo;Cut from&rdquo;: <em>Contig_stats.txt</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>Examine the new file by clicking on its <img src="../media/Galaxy-view.png" width=20 /> icon. We now have 2 columns instead of the 18 in the original file.</p>
<p><strong>2. Remove the Header lines of the new file.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Text Manipulation -&gt; Remove beginning</strong> and set the following:</li>
<li>&ldquo;Remove First&rdquo;: <em>1</em></li>
<li>&ldquo;from&rdquo;: <em>Cut on data1</em></li>
<li>click <strong>Execute</strong></li>
</ul>
<p>Note the the new file is the same as the previous one without the header line.</p>
<p><strong>3. Make a histogram.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Graph/Display Data -&gt; Histogram</strong> and set the following:</li>
<li>&ldquo;Dataset&rdquo;: <em>Remove beginning on Data X</em></li>
<li>&ldquo;Numerical column for X axis&rdquo;: <em>c2</em></li>
<li>&ldquo;Number of breaks&rdquo;: <em>25</em></li>
<li>&ldquo;Plot title&rdquo;: <em>Histogram of Contig Coverage</em></li>
<li>&ldquo;Label for X axis&rdquo;: <em>Coverage depth</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>Click on the <img src="../media/Galaxy-view.png" width=20 /> icon of the histogram to have a look at it. Note there are a few peaks. Maybe these correspond to single, double and triple copy number of these contigs.</p>
<p><strong>4. Calculate summary statistics for contig coverage depth.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Statistics -&gt; Summary Statisitics</strong> and set the following:</li>
<li>&ldquo;Summary statistics on&rdquo;: <em>Remove beginning on Data X</em></li>
<li>&ldquo;Column or expression&rdquo;: <em>c2</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<h3 id="example-2-convert-fastq-to-fasta">Example 2: Convert Fastq to Fasta<a class="headerlink" href="#example-2-convert-fastq-to-fasta" title="Permanent link">&para;</a></h3>
<p>This shows how to convert a fastq file to a fasta file. The tool creates a new file with the converted data.</p>
<p><strong>Converter tool</strong></p>
<ul>
<li>From the tool panel, click on <strong>Convert Formats -&gt; FASTQ to FASTA</strong> and set the following:</li>
<li>&ldquo;FASTQ file to convert&rdquo;: <em>Typical Fastq File</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>This will have created a new Fasta file called FASTQ to FASTA on data 2.</p>
<h3 id="example-3-find-ribosomal-rna-features-in-a-dna-sequence">Example 3: Find Ribosomal RNA Features in a DNA Sequence<a class="headerlink" href="#example-3-find-ribosomal-rna-features-in-a-dna-sequence" title="Permanent link">&para;</a></h3>
<p>This example shows how to use a tool called “barrnap” to search for rRNAs in a DNA sequence.</p>
<p><strong>1. Find all of the ribosomal RNA&rsquo;s in a sequence</strong></p>
<ul>
<li>From the tool panel, click on <strong>NGS: Annotation -&gt; barrnap</strong> and set the following:</li>
<li>&ldquo;Fasta file&rdquo;: MRSA252.fna</li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>A new file called <em>barrnap on data 3</em> will be produced. It is a gff3 file. (This stands for genome feature format - version 3. It is a file format for describing features contained by a DNA sequence.) Change its name to something more appropriate (click on the <img src="../media/Galaxy-edit.png" width=20 /> icon.)</p>
<p>Now let&rsquo;s say you only want the lines of the file for the 23S rRNA annotations. We can do this using a Filter tool.</p>
<p><strong>2. Filter the annotations to get the 23S RNAs</strong></p>
<ul>
<li>From the tool panel, click on <strong>Filter and Sort -&gt; Select</strong> and set the following:</li>
<li>&ldquo;Select lines from&rdquo;:  (whatever you called the barrnap gff3 output)</li>
<li>&ldquo;that&rdquo;: <em>Matching</em></li>
<li>&ldquo;the pattern&rdquo;: <em>23S</em>     (this will look for all the lines in the file that contain “23S”)</li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>Now you have a gff3 file with just the 23S annotations!</p>
<h2 id="what-now">What now?<a class="headerlink" href="#what-now" title="Permanent link">&para;</a></h2>
<p>Remember how we started a new history at the beginning? If you want to see any of your old histories, click on the history menu button <img src="../media/Galaxy-menu.png" width=20 /> at the top of the histories panel and then select “Saved Histories.” This will give you a list of all the histories you have worked on in this Galaxy server.</p>
<p>That&rsquo;s it. You now know a bit about the Galaxy interface and how to load data, run tools and view their outputs. For more tutorials, see <a href="http://genome.edu.au/learn">http://genome.edu.au/learn</a></p>
                
              
              
                


              
            </article>
          </div>
        </div>
        
      </main>
      
        
<footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            <a href="https://www.melbournebioinformatics.org.au/">Melbourne Bioinformatics</a>
          </div>
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
        
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../..", "features": [], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../../assets/javascripts/workers/search.b0710199.min.js", "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.76f349be.min.js"></script>
      
    
  </body>
</html>