
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.2.1, mkdocs-material-7.1.8">
    
    
      
        <title>Introduction to Galaxy Workflows - Bioinformatics Documentation</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.ca7ac06f.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.f1a3b89f.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font-family:"Roboto";--md-code-font-family:"Roboto Mono"}</style>
      
    
    
    
      <link rel="stylesheet" href="../../extra.css">
    
    
<script async src="https://www.googletagmanager.com/gtag/js?id=G-EMNLJD0DWP"></script>
<script>
    window.dataLayer = window.dataLayer || [];
    function gtag() {
        dataLayer.push(arguments);
    }
    gtag('js', new Date());
    gtag('config', 'G-EMNLJD0DWP');
</script>

    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
  
    
    <script>function __prefix(e){return new URL("../..",location).pathname+"."+e}function __get(e,t=localStorage){return JSON.parse(t.getItem(__prefix(e)))}</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#galaxy-workflows" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../index.html" title="Bioinformatics Documentation" class="md-header__button md-logo" aria-label="Bioinformatics Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Bioinformatics Documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Introduction to Galaxy Workflows
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        
<a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    melbournebioinformatics/MelBioInf_docs
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../index.html" title="Bioinformatics Documentation" class="md-nav__button md-logo" aria-label="Bioinformatics Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    Bioinformatics Documentation
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    melbournebioinformatics/MelBioInf_docs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../index.html" class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" type="checkbox" id="__nav_2" checked>
      
      <label class="md-nav__link" for="__nav_2">
        Tools and skill development
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Tools and skill development" data-md-level="1">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Tools and skill development
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../guides/bioinfo.html" class="md-nav__link">
        What is Bioinformatics?
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../guides/galaxy.html" class="md-nav__link">
        Introduction to the Galaxy Platform
      </a>
    </li>
  

          
            
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Introduction to Galaxy Workflows
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="galaxy-workflows.html" class="md-nav__link md-nav__link--active">
        Introduction to Galaxy Workflows
      </a>
      
        
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#background" class="md-nav__link">
    Background
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-1-preparation" class="md-nav__link">
    Section 1: Preparation.
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-2-create-and-run-a-workflow" class="md-nav__link">
    Section 2: Create and run a workflow.
  </a>
  
    <nav class="md-nav" aria-label="Section 2: Create and run a workflow.">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#import-the-workflow-history" class="md-nav__link">
    Import the workflow history
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#workflow-creation-method-1" class="md-nav__link">
    Workflow creation: Method 1
  </a>
  
    <nav class="md-nav" aria-label="Workflow creation: Method 1">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#some-discussion" class="md-nav__link">
    Some discussion
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#workflow-creation-method-2" class="md-nav__link">
    Workflow Creation: Method 2
  </a>
  
    <nav class="md-nav" aria-label="Workflow Creation: Method 2">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-create-a-workflow-name-and-edit-space" class="md-nav__link">
    Step 1: Create a workflow name and edit space.
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-open-the-editor-and-place-component-tools" class="md-nav__link">
    Step 2: Open the editor and place component tools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-save-it" class="md-nav__link">
    Step 3: Save it!
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-workflow" class="md-nav__link">
    Running the workflow
  </a>
  
    <nav class="md-nav" aria-label="Running the workflow">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#create-the-new-history" class="md-nav__link">
    Create the new history
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#run-the-workflow" class="md-nav__link">
    Run the workflow
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#what-now" class="md-nav__link">
    What now?
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../python_overview/python_overview.html" class="md-nav__link">
        Introduction to Python
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../using_git/Using_Git.html" class="md-nav__link">
        Introduction to Git and Github
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../hpc/hpc.html" class="md-nav__link">
        Introduction to HPC
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../unix/unix.html" class="md-nav__link">
        Introduction to Unix
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../intro_R_biologists/intro_R_biologists.html" class="md-nav__link">
        Introduction to R for Biologists
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../Genome_browsers/GenomeBrowsers_Intro.html" class="md-nav__link">
        Introduction to Genome Browsers
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_exp_design/rna_seq_experimental_design.html" class="md-nav__link">
        RNA-seq Experimental Design
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../docker/docker.html" class="md-nav__link">
        Containerized Bioinformatics
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../cwl/cwl.html" class="md-nav__link">
        Common Workflow Language for Bioinformatics
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../bionitio/bionitio.html" class="md-nav__link">
        Command-line software development
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" >
      
      <label class="md-nav__link" for="__nav_3">
        Genomics
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Genomics" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Genomics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_1" type="checkbox" id="__nav_3_1" >
      
      <label class="md-nav__link" for="__nav_3_1">
        Variant detection
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Variant detection" data-md-level="2">
        <label class="md-nav__title" for="__nav_3_1">
          <span class="md-nav__icon md-icon"></span>
          Variant detection
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../var_detect_advanced/var_detect_advanced_background.html" class="md-nav__link">
        Introduction to Variant detection
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling_galaxy_1/variant_calling_galaxy_1.html" class="md-nav__link">
        Variant Calling part 1 (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../var_detect_advanced/var_detect_advanced.html" class="md-nav__link">
        Variant Calling part 2 (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../longread_sv_calling/longread_sv_calling.html" class="md-nav__link">
        Long-read Structural Variant Calling
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_2" type="checkbox" id="__nav_3_2" >
      
      <label class="md-nav__link" for="__nav_3_2">
        de novo assembly
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="de novo assembly" data-md-level="2">
        <label class="md-nav__title" for="__nav_3_2">
          <span class="md-nav__icon md-icon"></span>
          de novo assembly
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../assembly/assembly-background.html" class="md-nav__link">
        Introduction to de novo assembly with Velvet
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../assembly/assembly-protocol.html" class="md-nav__link">
        Introduction to de novo genome assembly for Illumina reads
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../assembly/assembly.html" class="md-nav__link">
        de novo assembly of Illumina reads using Velvet (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../assembly/spades.html" class="md-nav__link">
        de novo assembly of Illumina reads using Spades (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../pacbio/index.html" class="md-nav__link">
        de novo assembly of PacBio reads (unix)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../hybrid_assembly/nanopore_assembly.html" class="md-nav__link">
        Hybrid genome assembly - Nanopore and Illumina
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" >
      
      <label class="md-nav__link" for="__nav_4">
        Transcriptomics
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Transcriptomics" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          Transcriptomics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_exp_design/rna_seq_experimental_design.html" class="md-nav__link">
        RNAseq Experimental Design
      </a>
    </li>
  

          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_2" type="checkbox" id="__nav_4_2" >
      
      <label class="md-nav__link" for="__nav_4_2">
        Bulk RNAseq
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Bulk RNAseq" data-md-level="2">
        <label class="md-nav__title" for="__nav_4_2">
          <span class="md-nav__icon md-icon"></span>
          Bulk RNAseq
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_dge_basic/rna_seq_basic_background.html" class="md-nav__link">
        Introduction to bulk RNAseq analysis
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_dge_basic/rna_seq_basic_tutorial.html" class="md-nav__link">
        RNAseq analysis using HISAT2 (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_dge_basic/rna_seq_basic_tuxedo.html" class="md-nav__link">
        RNAseq analysis using Tuxedo (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_dge_advanced/rna_seq_advanced_tutorial.html" class="md-nav__link">
        RNAseq differential expression tool comparision (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../rna_seq_dge_in_r/rna_seq_r.html" class="md-nav__link">
        RNAseq differential expression in R
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_3" type="checkbox" id="__nav_4_3" >
      
      <label class="md-nav__link" for="__nav_4_3">
        Singlecell RNAseq
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Singlecell RNAseq" data-md-level="2">
        <label class="md-nav__title" for="__nav_4_3">
          <span class="md-nav__icon md-icon"></span>
          Singlecell RNAseq
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../singlecell/index.html" class="md-nav__link">
        10X single-cell RNA-seq analysis in R
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" >
      
      <label class="md-nav__link" for="__nav_5">
        Proteomics
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Proteomics" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Proteomics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../proteomics_basic/proteomics_basic.html" class="md-nav__link">
        Identifying proteins from mass spectrometry data
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../proteomics_basic/background_data_formats.html" class="md-nav__link">
        Data Formats Background
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../proteomics_basic/background_protein_databases.html" class="md-nav__link">
        Protein Databases
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../proteomics_basic/background_search_engines.html" class="md-nav__link">
        Search Engines
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../proteomics_basic/background_protein_prophet.html" class="md-nav__link">
        Protein Prophet
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" >
      
      <label class="md-nav__link" for="__nav_6">
        Statistics and Visualisation
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Statistics and Visualisation" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Statistics and Visualisation
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../python_viz/python_viz.html" class="md-nav__link">
        Data visualisation with Python
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_7" type="checkbox" id="__nav_7" >
      
      <label class="md-nav__link" for="__nav_7">
        Structural Modelling
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Structural Modelling" data-md-level="1">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          Structural Modelling
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../molecular_dynamics_101/molecular_dynamics_101.html" class="md-nav__link">
        Molecular Dynamics - Introduction to cluster computing
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../molecular_dynamics_201/molecular_dynamics_201.html" class="md-nav__link">
        Molecular Dynamics - Building input files, visualising the trajectory
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#background" class="md-nav__link">
    Background
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-1-preparation" class="md-nav__link">
    Section 1: Preparation.
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-2-create-and-run-a-workflow" class="md-nav__link">
    Section 2: Create and run a workflow.
  </a>
  
    <nav class="md-nav" aria-label="Section 2: Create and run a workflow.">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#import-the-workflow-history" class="md-nav__link">
    Import the workflow history
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#workflow-creation-method-1" class="md-nav__link">
    Workflow creation: Method 1
  </a>
  
    <nav class="md-nav" aria-label="Workflow creation: Method 1">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#some-discussion" class="md-nav__link">
    Some discussion
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#workflow-creation-method-2" class="md-nav__link">
    Workflow Creation: Method 2
  </a>
  
    <nav class="md-nav" aria-label="Workflow Creation: Method 2">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-create-a-workflow-name-and-edit-space" class="md-nav__link">
    Step 1: Create a workflow name and edit space.
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-open-the-editor-and-place-component-tools" class="md-nav__link">
    Step 2: Open the editor and place component tools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-save-it" class="md-nav__link">
    Step 3: Save it!
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#running-the-workflow" class="md-nav__link">
    Running the workflow
  </a>
  
    <nav class="md-nav" aria-label="Running the workflow">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#create-the-new-history" class="md-nav__link">
    Create the new history
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#run-the-workflow" class="md-nav__link">
    Run the workflow
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#what-now" class="md-nav__link">
    What now?
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/melbournebioinformatics/MelBioInf_docs/edit/master/docs/tutorials/galaxy-workflows/galaxy-workflows.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                <style>
  .image-header-gvl {
    -webkit-column-count: 3;
    -moz-column-count: 3;
    column-count: 3;

    -webkit-column-gap: 140px;
    -moz-column-gap: 140px;
    column-gap: 140px;
  }
</style>
<div class="image-header-gvl">
    <a href=http://genome.edu.au><img src="../media/gvl_logo.jpg" alt="GVL logo" width="112"/></a>
    <a href="http://galaxyproject.org"><img src="../media/GalaxyLogoHighRes.png" width="164" /></a>
    <a href="https://www.melbournebioinformatics.org.au/"><img src="../../../img/melbioinf_logo.png" alt="Melbourne Bioinformatics logo" width="164"/></a>
</div>

<h1 id="galaxy-workflows">Galaxy Workflows<a class="headerlink" href="#galaxy-workflows" title="Permanent link">&para;</a></h1>
<p>Written and maintained by <a href="mailto:simon.gladman@unimelb.edu.au">Simon Gladman</a> - Melbourne Bioinformatics (formerly VLSCI)</p>
<h2 id="background">Background<a class="headerlink" href="#background" title="Permanent link">&para;</a></h2>
<p>This workshop/tutorial will familiarise you with the Galaxy workflow engine. It will cover the following topics:</p>
<ul>
<li>Logging in to the server</li>
<li>How to construct and use a workflow by various methods</li>
<li>How to share a workflow</li>
</ul>
<hr />
<h2 id="section-1-preparation">Section 1: Preparation.<a class="headerlink" href="#section-1-preparation" title="Permanent link">&para;</a></h2>
<p>The purpose of this section is to get you to log in to the server.. For this workshop, you can use a Galaxy server that you have created by following the steps outlined in: <a href="">Launch a GVL Galaxy Instance</a> or you can use the public <a href="https://usegalaxy.org.au/">Galaxy Australia</a></p>
<p><strong>Go to Galaxy URL of your server in Firefox or Chrome (your choice, please don&rsquo;t use IE or Safari.)</strong></p>
<ul>
<li>If you have previously registered on this server just log in:<ul>
<li>On the top menu select: <strong>User -&gt; Login</strong></li>
<li>Enter your password</li>
<li>Click the <strong>Submit</strong> button</li>
</ul>
</li>
</ul>
<!-- -->

<ul>
<li>If you haven’t registered on this server, you’ll need to now.<ul>
<li>On the top menu select: <strong>User -&gt; Register</strong></li>
<li>Enter your email, choose a password, repeat it and add a (all lower case) one word name</li>
<li>Click on the <strong>Submit</strong> button.</li>
</ul>
</li>
</ul>
<hr />
<h2 id="section-2-create-and-run-a-workflow">Section 2: Create and run a workflow.<a class="headerlink" href="#section-2-create-and-run-a-workflow" title="Permanent link">&para;</a></h2>
<p>This section will show you two different methods to create a workflow and then how to run one.</p>
<h3 id="import-the-workflow-history">Import the workflow history<a class="headerlink" href="#import-the-workflow-history" title="Permanent link">&para;</a></h3>
<p>In this step we will import a shared history to our workspace so we can extract a workflow from it. This will only work on Galaxy servers which have the history available on it. If yours doesn&rsquo;t have the appropriate history, there are instructions to create it <a href="history_creation.html">here</a>. Alternatively, you can extract a workflow from any history you have in your &ldquo;Saved Histories.&rdquo;</p>
<ul>
<li>From the menu at the top of the Galaxy window, click <strong>Shared Data -&gt; Histories</strong></li>
<li>Find the history called &ldquo;<em>workflow_finished</em>&rdquo; and click on it.</li>
<li>Then click on <strong>Import history</strong> at the top right of the screen.</li>
<li>Change the name if you wish and then click <strong>Import</strong></li>
</ul>
<p>This history should now be in your history pane on the right.</p>
<!--


This step will show you how to import a history from a remote source into your own workspace. We will be using this history to build a workflow.

* From the histories menu <img src="../media/Galaxy-menu.png" width=20 />, click **Import from File**.
* Enter *https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/Workflow-finished-history.tar.gz* in the URL box.
* Click **Submit**

You might have to wait for a bit, then:

* From the history menu <img src="../media/Galaxy-menu.png" width=20 />, click **Saved Histories**
* Select the history: *Imported: Workflow-finished*


-->
<h2 id="workflow-creation-method-1">Workflow creation: Method 1<a class="headerlink" href="#workflow-creation-method-1" title="Permanent link">&para;</a></h2>
<p>We will create a workflow from an existing history. You can use this method to make a re-useable analysis from one you’ve already done. i.e. You can perform the analysis once and then create a workflow out of it to re-use it on more/new data. We will create a workflow from the history you imported in step 1 above. The footnote below explains the steps that created this history. These are the steps we will mimic in the workflow.</p>
<p>Make sure your current history is the one you imported in Section 2 - Step 1 above (<em>imported: workflow_finished.</em>) If not, switch to it. If you couldn&rsquo;t import it, you should be able to complete this step with any suitable history.</p>
<p><strong>Now we will create the workflow.</strong></p>
<ul>
<li>Click on the histories menu button <img src="../media/Galaxy-menu.png" width=20 /> at the top of the history pane.</li>
<li>Click <strong>Extract Workflow</strong></li>
</ul>
<p>You will now be shown a page which contains the steps used to create the history you are extracting from. We use this page to say what to include in the workflow. We want everything here so we’ll just accept the defaults and:</p>
<ul>
<li>Change the Workflow name to something sensible like “Basic Variant Calling Workflow”</li>
<li>Click <strong>Create Workflow</strong></li>
</ul>
<p>The workflow is now accessible via the bottom of the tool pane by clicking on All Workflows.</p>
<h3 id="some-discussion">Some discussion<a class="headerlink" href="#some-discussion" title="Permanent link">&para;</a></h3>
<p>Have a look at your workflow. Click on its button in the workflow list. It’s tool interface will appear. You can now run this workflow any time you like with different input datasets. NOTE: The input data sets must be of the same types as the original ones. i.e. In our case, two fastq reads files and one fasta reference sequence.</p>
<p>More interesting though is to:</p>
<ul>
<li>Click on the  <strong>Workflows</strong> link in the top menu</li>
<li>Click on the down arrow on your workflow’s button.</li>
<li>Click <strong>Edit</strong></li>
</ul>
<p>A visualisation of your workflow will appear. Note the connections and the steps.</p>
<p>Next we’ll go through how to create this workflow using the editor..</p>
<h2 id="workflow-creation-method-2">Workflow Creation: Method 2<a class="headerlink" href="#workflow-creation-method-2" title="Permanent link">&para;</a></h2>
<p>We will now create the same read mapping/variant calling workflow using the editor directly. The workflow needs to take in some reads and a reference, map the reads to the reference using BWA, run Freebayes on the BAM output from BWA to call the variants, and finally filter the resulting vcf file.</p>
<h3 id="step-1-create-a-workflow-name-and-edit-space">Step 1: Create a workflow name and edit space.<a class="headerlink" href="#step-1-create-a-workflow-name-and-edit-space" title="Permanent link">&para;</a></h3>
<ul>
<li>Click on <strong>Workflow</strong> in Galaxy’s menu.</li>
<li>Click on the <strong>Create New Workflow</strong> button.</li>
<li>In the &ldquo;Workflow Name&rdquo; text box type: <em>Variants from scratch</em></li>
<li>Click the <strong>Create</strong> button.</li>
</ul>
<p>You should now be presented with a blank workflow editing grid.</p>
<h3 id="step-2-open-the-editor-and-place-component-tools">Step 2: Open the editor and place component tools<a class="headerlink" href="#step-2-open-the-editor-and-place-component-tools" title="Permanent link">&para;</a></h3>
<p><strong>Add three input datafiles.</strong></p>
<ul>
<li>In the Workflow control section of the tool pane, click on <strong>Inputs -&gt; Input dataset</strong> three times.</li>
<li>Spread them out towards the left hand side of the workflow grid by clicking and dragging them around.</li>
<li>For each one, change their name.<ul>
<li>Click on each input box in turn</li>
<li>In the right hand pane (where the history usually is), change the name to:<ul>
<li>Reference data</li>
<li>Reads 1</li>
<li>Reads 2 - respectively.</li>
</ul>
</li>
<li>If you click on <strong>Input dataset</strong> in the tan box at the top of the right hand panel on the screen, you can change the name of the box as it appears on the editing layout. You need to give each one a more sensible name and press <em>Enter</em> to make the change.</li>
</ul>
</li>
</ul>
<p><strong>Add in the BWA mapping step.</strong></p>
<ul>
<li>Click on <strong>NGS: Mapping -&gt; Map with BWA</strong> in the tool pane. BWA will be added to the workflow grid.</li>
<li>In the right hand pane (where the history usually is), change the following parameters.<ul>
<li>Change “Will you select a reference genome from your history or use a built-in index?:” to <em>Use a genome from history.</em></li>
<li>Note that the BWA box on the grid changes to match these settings.</li>
</ul>
</li>
</ul>
<p><strong>Connect the tools together.</strong></p>
<ul>
<li>Click and drag on the output of one of the input dataset tools to each of the input spots in the BWA tool. (Make connections.)<ul>
<li>&ldquo;Reference data&rdquo; output to reference to &ldquo;Use the following dataset as the reference sequence&rdquo;</li>
<li>&ldquo;Reads 1&rdquo; output to &ldquo;Select first set of reads&rdquo;</li>
<li>&ldquo;Reads 2&rdquo; output to &ldquo;Select second set of reads&rdquo;</li>
</ul>
</li>
</ul>
<p><strong>Add in the Freebayes (Variant Calling step.)</strong></p>
<ul>
<li>Click on <strong>NGS: Variant Calling -&gt; Freebayes</strong></li>
<li>In the right hand pane, change the following:<ul>
<li>&ldquo;Choose the source for the reference list:&rdquo;  to <em>History</em></li>
<li>Connect &ldquo;Map with BWA&rdquo; bam output to &ldquo;Freebayes’&rdquo; bam input.</li>
<li>Connect the &ldquo;Reference data&rdquo; output to &ldquo;Freebayes’&rdquo; <em>Use the following dataset as the reference sequence</em> input.</li>
</ul>
</li>
</ul>
<p>If you’re keen - <strong>Note: this is optional.</strong> Also change the following parameters the right hand pane for freebayes to make it a bit more sensible for variant calling in bacterial genomes.</p>
<ul>
<li>&ldquo;Choose parameter selection level&rdquo;: <em>Complete list of all options</em></li>
<li>&ldquo;Population model options&rdquo;: <em>Set population model options</em></li>
<li>&ldquo;Set ploidy for the analysis&rdquo;: <em>1</em></li>
<li>&ldquo;Input filters&rdquo;: <em>Set input filters</em></li>
<li>&ldquo;Exclude alignments from analysis if they have a mapping quality less than&rdquo;: <em>20</em></li>
<li>&ldquo;Exclude alleles from analysis if their supporting base quality is less than&rdquo;: <em>20</em></li>
<li>&ldquo;Require at least this fraction of observations … to evaluate the position&rdquo;: <em>0.9</em></li>
<li>&ldquo;Require at least this count of observations .. to evaluate the position&rdquo;: <em>10</em></li>
<li>&ldquo;Population and mappability priors&rdquo;: <em>Set population and mappability priors</em></li>
<li>&ldquo;Disable incorporation of prior expectations about observations&rdquo;: <em>Yes</em></li>
</ul>
<p><strong>Add in the Filter step.</strong></p>
<ul>
<li>Click on <strong>Filter and Sort - &gt; Filter</strong></li>
<li>Connect &ldquo;Freebayes’&rdquo; output_vcf to the &ldquo;filter&rdquo; input.</li>
<li>In the right hand pane, change the following:<ul>
<li>&ldquo;With the following condition&rdquo;: <em>c6 &gt; 500</em></li>
<li>&ldquo;Number of header lines to skip&rdquo;: <em>56</em></li>
</ul>
</li>
</ul>
<p>Phew! We’re nearly done! The only thing left is to select which workflow outputs we want to keep in our history. Next to each output for every tool is a star. Clicking on the stars will select those files as workflow outputs, everything else will be hidden in the history. In this case we only really want the BAM file and the final variants (vcf file.) Therefore:</p>
<p><strong>Select workflow outputs.</strong></p>
<ul>
<li>Click on the star next to &ldquo;Map with BWA’s&rdquo; bam file output.</li>
<li>Click on the star next to &ldquo;Filter’s&rdquo; output vcf.</li>
</ul>
<h3 id="step-3-save-it">Step 3: Save it!<a class="headerlink" href="#step-3-save-it" title="Permanent link">&para;</a></h3>
<p>Click on the <img src="../media/Galaxy-menu.png" width=20 /> at the top of the workflow grid and select <strong>Save</strong>.</p>
<p>Congratulations. You’ve just created a Galaxy workflow.</p>
<p>Now to run it!</p>
<h2 id="running-the-workflow">Running the workflow<a class="headerlink" href="#running-the-workflow" title="Permanent link">&para;</a></h2>
<p>We will now make a new history called &ldquo;Test&rdquo; and run the workflow on it’s data.</p>
<h3 id="create-the-new-history">Create the new history<a class="headerlink" href="#create-the-new-history" title="Permanent link">&para;</a></h3>
<ul>
<li>From the Histories Menu, select <strong>Copy Datasets</strong></li>
<li>Select the 2 x fastq files and the Ecoli .fna file.</li>
<li>Under destination history, enter <em>Test</em> into &ldquo;New History Named:&rdquo;</li>
<li>Click <strong>Copy History Items</strong> button</li>
<li>Click on the link to the new history in the green bar at the top of the screen</li>
</ul>
<h3 id="run-the-workflow">Run the workflow<a class="headerlink" href="#run-the-workflow" title="Permanent link">&para;</a></h3>
<ul>
<li>On the tools pane, click <strong>All Workflows</strong></li>
<li>Select the <em>Variants from scratch</em> workflow (or whatever you called it.)</li>
<li>Give it the correct files.</li>
<li><em>Ecoli &hellip; .fna</em> for Reference data</li>
<li><em>bacterial_std_err_1.fastq</em> for Reads 1</li>
<li><em>bacterial_std_err_2.fastq</em> for Reads 2</li>
<li>Click <strong>Run Workflow</strong></li>
</ul>
<p>Your workflow will now run. It will send the right files to the right tools at the right time to the cluster (compute engine on your machine) and wait for them to finish. Watch as they turn yellow then green in turn.</p>
<h2 id="what-now">What now?<a class="headerlink" href="#what-now" title="Permanent link">&para;</a></h2>
<p>Where to start? There’s so much you can do with Workflows. You can even run them on multiple file sets from the one setup.</p>
                
              
              
                


              
            </article>
          </div>
        </div>
        
      </main>
      
        
<footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="../../guides/galaxy.html" class="md-footer__link md-footer__link--prev" aria-label="Previous: Introduction to the Galaxy Platform" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Introduction to the Galaxy Platform
            </div>
          </div>
        </a>
      
      
        
        <a href="../python_overview/python_overview.html" class="md-footer__link md-footer__link--next" aria-label="Next: Introduction to Python" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Introduction to Python
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            <a href="https://www.melbournebioinformatics.org.au/">Melbourne Bioinformatics</a>
          </div>
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
        
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../..", "features": [], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../../assets/javascripts/workers/search.b0710199.min.js", "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.76f349be.min.js"></script>
      
    
  </body>
</html>