
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.1.2, mkdocs-material-7.0.6">
    
    
      
        <title>Introduction to Variant detection - Bioinformatics Documentation</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.2c0c5eaf.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.7fa14f5b.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font-family:"Roboto";--md-code-font-family:"Roboto Mono"}</style>
      
    
    
    
      <link rel="stylesheet" href="../../../extra.css">
    
    
      
        
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-21794901-5","www.melbournebioinformatics.org.au/tutorials/"),ga("set","anonymizeIp",!0),ga("send","pageview"),document.addEventListener("DOMContentLoaded",function(){document.forms.search&&document.forms.search.query.addEventListener("blur",function(){var e;this.value&&(e=document.location.pathname,ga("send","pageview",e+"?q="+this.value))}),"undefined"!=typeof location$&&location$.subscribe(function(e){ga("send","pageview",e.pathname)})})</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
      
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
      
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#introduction-to-variant-detection" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="Bioinformatics Documentation" class="md-header__button md-logo" aria-label="Bioinformatics Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Bioinformatics Documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Introduction to Variant detection
            
          </span>
        </div>
      </div>
    </div>
    <div class="md-header__options">
      
    </div>
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" data-md-state="active" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        
<a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    melbournebioinformatics/MelBioInf_docs
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="Bioinformatics Documentation" class="md-nav__button md-logo" aria-label="Bioinformatics Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    Bioinformatics Documentation
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/melbournebioinformatics/MelBioInf_docs/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05L244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    melbournebioinformatics/MelBioInf_docs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" type="checkbox" id="__nav_2" >
      
      <label class="md-nav__link" for="__nav_2">
        Tools and skill development
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Tools and skill development" data-md-level="1">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Tools and skill development
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/bioinfo/" class="md-nav__link">
        What is Bioinformatics?
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/galaxy/" class="md-nav__link">
        Introduction to the Galaxy Platform
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../galaxy-workflows/galaxy-workflows/" class="md-nav__link">
        Introduction to Galaxy Workflows
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../python_overview/python_overview/" class="md-nav__link">
        Introduction to Python
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../using_git/Using_Git/" class="md-nav__link">
        Introduction to Git and Github
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../hpc/hpc/" class="md-nav__link">
        Introduction to HPC
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../unix/unix/" class="md-nav__link">
        Introduction to Unix
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../intro_R_biologists/intro_R_biologists/" class="md-nav__link">
        Introduction to R for Biologists
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../Genome_browsers/GenomeBrowsers_Intro/" class="md-nav__link">
        Introduction to Genome Browsers
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../rna_seq_exp_design/rna_seq_experimental_design/" class="md-nav__link">
        RNA-seq Experimental Design
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../docker/docker/" class="md-nav__link">
        Containerized Bioinformatics
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../cwl/cwl/" class="md-nav__link">
        Common Workflow Language for Bioinformatics
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../bionitio/bionitio/" class="md-nav__link">
        Command-line software development
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" checked>
      
      <label class="md-nav__link" for="__nav_3">
        Genomics
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Genomics" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Genomics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_1" type="checkbox" id="__nav_3_1" checked>
      
      <label class="md-nav__link" for="__nav_3_1">
        Variant detection
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Variant detection" data-md-level="2">
        <label class="md-nav__title" for="__nav_3_1">
          <span class="md-nav__icon md-icon"></span>
          Variant detection
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Introduction to Variant detection
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Introduction to Variant detection
      </a>
      
        
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#background" class="md-nav__link">
    Background
  </a>
  
    <nav class="md-nav" aria-label="Background">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#variant-calling-vs-genotyping" class="md-nav__link">
    Variant Calling vs genotyping
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#process-of-variant-calling" class="md-nav__link">
    Process of variant calling
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#homozygous-or-heterozygous-mutations" class="md-nav__link">
    Homozygous or Heterozygous mutations:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-calling-software" class="md-nav__link">
    Variant Calling Software:
  </a>
  
    <nav class="md-nav" aria-label="Variant Calling Software:">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#variant-calling-using-samtools-mpileup-bcftools" class="md-nav__link">
    Variant Calling using Samtools (Mpileup + bcftools)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variant-calling-using-gatk-unified-genotyper" class="md-nav__link">
    Variant Calling using GATK-Unified Genotyper
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variant-calling-using-freebayes" class="md-nav__link">
    Variant Calling using FreeBayes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#evaluation-of-detected-variants-using-variant-eval" class="md-nav__link">
    Evaluation of detected variants using Variant Eval
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-matrix" class="md-nav__link">
    Quality Matrix:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#local-realignment" class="md-nav__link">
    Local realignment
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-galaxy-workflow-platform" class="md-nav__link">
    The Galaxy workflow platform
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-format-used-in-the-tutorial" class="md-nav__link">
    Data Format used in the tutorial
  </a>
  
    <nav class="md-nav" aria-label="Data Format used in the tutorial">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sequence-alignment-map-format" class="md-nav__link">
    Sequence Alignment Map format
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#binary-sequence-alignment-map-format" class="md-nav__link">
    Binary Sequence Alignment Map format
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vcf-file-format" class="md-nav__link">
    VCF file format
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vcf-format-in-galaxy" class="md-nav__link">
    VCF format in Galaxy:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bcf-file-format" class="md-nav__link">
    Bcf file format:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#_1" class="md-nav__link">
    
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../variant_calling_galaxy_1/variant_calling_galaxy_1/" class="md-nav__link">
        Variant Calling part 1 (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../var_detect_advanced/" class="md-nav__link">
        Variant Calling part 2 (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../longread_sv_calling/longread_sv_calling/" class="md-nav__link">
        Long-read Structural Variant Calling
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_2" type="checkbox" id="__nav_3_2" >
      
      <label class="md-nav__link" for="__nav_3_2">
        de novo assembly
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="de novo assembly" data-md-level="2">
        <label class="md-nav__title" for="__nav_3_2">
          <span class="md-nav__icon md-icon"></span>
          de novo assembly
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../assembly/assembly-background/" class="md-nav__link">
        Introduction to de novo assembly with Velvet
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../assembly/assembly-protocol/" class="md-nav__link">
        Introduction to de novo genome assembly for Illumina reads
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../assembly/assembly/" class="md-nav__link">
        de novo assembly of Illumina reads using Velvet (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../assembly/spades/" class="md-nav__link">
        de novo assembly of Illumina reads using Spades (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../pacbio/" class="md-nav__link">
        de novo assembly of PacBio reads (unix)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../hybrid_assembly/nanopore_assembly/" class="md-nav__link">
        Hybrid genome assembly - Nanopore and Illumina
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" >
      
      <label class="md-nav__link" for="__nav_4">
        Transcriptomics
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Transcriptomics" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          Transcriptomics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../rna_seq_exp_design/rna_seq_experimental_design/" class="md-nav__link">
        RNAseq Experimental Design
      </a>
    </li>
  

          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_2" type="checkbox" id="__nav_4_2" >
      
      <label class="md-nav__link" for="__nav_4_2">
        Bulk RNAseq
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Bulk RNAseq" data-md-level="2">
        <label class="md-nav__title" for="__nav_4_2">
          <span class="md-nav__icon md-icon"></span>
          Bulk RNAseq
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../rna_seq_dge_basic/rna_seq_basic_background/" class="md-nav__link">
        Introduction to bulk RNAseq analysis
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../rna_seq_dge_basic/rna_seq_basic_tutorial/" class="md-nav__link">
        RNAseq analysis using HISAT2 (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../rna_seq_dge_basic/rna_seq_basic_tuxedo/" class="md-nav__link">
        RNAseq analysis using Tuxedo (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../rna_seq_dge_advanced/rna_seq_advanced_tutorial/" class="md-nav__link">
        RNAseq differential expression tool comparision (Galaxy)
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../rna_seq_dge_in_r/rna_seq_r/" class="md-nav__link">
        RNAseq differential expression in R
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
            
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_3" type="checkbox" id="__nav_4_3" >
      
      <label class="md-nav__link" for="__nav_4_3">
        Singlecell RNAseq
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Singlecell RNAseq" data-md-level="2">
        <label class="md-nav__title" for="__nav_4_3">
          <span class="md-nav__icon md-icon"></span>
          Singlecell RNAseq
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../singlecell/" class="md-nav__link">
        10X single-cell RNA-seq analysis in R
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" >
      
      <label class="md-nav__link" for="__nav_5">
        Proteomics
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Proteomics" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Proteomics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../proteomics_basic/proteomics_basic/" class="md-nav__link">
        Identifying proteins from mass spectrometry data
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../proteomics_basic/background_data_formats/" class="md-nav__link">
        Data Formats Background
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../proteomics_basic/background_protein_databases/" class="md-nav__link">
        Protein Databases
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../proteomics_basic/background_search_engines/" class="md-nav__link">
        Search Engines
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../proteomics_basic/background_protein_prophet/" class="md-nav__link">
        Protein Prophet
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" >
      
      <label class="md-nav__link" for="__nav_6">
        Statistics and Visualisation
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Statistics and Visualisation" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Statistics and Visualisation
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../python_viz/python_viz/" class="md-nav__link">
        Data visualisation with Python
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_7" type="checkbox" id="__nav_7" >
      
      <label class="md-nav__link" for="__nav_7">
        Structural Modelling
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Structural Modelling" data-md-level="1">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          Structural Modelling
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../molecular_dynamics_101/molecular_dynamics_101/" class="md-nav__link">
        Molecular Dynamics - Introduction to cluster computing
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../molecular_dynamics_201/molecular_dynamics_201/" class="md-nav__link">
        Molecular Dynamics - Building input files, visualising the trajectory
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#background" class="md-nav__link">
    Background
  </a>
  
    <nav class="md-nav" aria-label="Background">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#variant-calling-vs-genotyping" class="md-nav__link">
    Variant Calling vs genotyping
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#process-of-variant-calling" class="md-nav__link">
    Process of variant calling
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#homozygous-or-heterozygous-mutations" class="md-nav__link">
    Homozygous or Heterozygous mutations:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-calling-software" class="md-nav__link">
    Variant Calling Software:
  </a>
  
    <nav class="md-nav" aria-label="Variant Calling Software:">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#variant-calling-using-samtools-mpileup-bcftools" class="md-nav__link">
    Variant Calling using Samtools (Mpileup + bcftools)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variant-calling-using-gatk-unified-genotyper" class="md-nav__link">
    Variant Calling using GATK-Unified Genotyper
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#variant-calling-using-freebayes" class="md-nav__link">
    Variant Calling using FreeBayes
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#evaluation-of-detected-variants-using-variant-eval" class="md-nav__link">
    Evaluation of detected variants using Variant Eval
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#quality-matrix" class="md-nav__link">
    Quality Matrix:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#local-realignment" class="md-nav__link">
    Local realignment
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#the-galaxy-workflow-platform" class="md-nav__link">
    The Galaxy workflow platform
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-format-used-in-the-tutorial" class="md-nav__link">
    Data Format used in the tutorial
  </a>
  
    <nav class="md-nav" aria-label="Data Format used in the tutorial">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#sequence-alignment-map-format" class="md-nav__link">
    Sequence Alignment Map format
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#binary-sequence-alignment-map-format" class="md-nav__link">
    Binary Sequence Alignment Map format
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vcf-file-format" class="md-nav__link">
    VCF file format
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#vcf-format-in-galaxy" class="md-nav__link">
    VCF format in Galaxy:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#bcf-file-format" class="md-nav__link">
    Bcf file format:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#_1" class="md-nav__link">
    
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/melbournebioinformatics/MelBioInf_docs/edit/master/docs/tutorials/var_detect_advanced/var_detect_advanced_background.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                <h1 id="introduction-to-variant-detection">Introduction to Variant detection<a class="headerlink" href="#introduction-to-variant-detection" title="Permanent link">&para;</a></h1>
<h2 id="background">Background<a class="headerlink" href="#background" title="Permanent link">&para;</a></h2>
<p>A variant is something that is different from a standard or type.</p>
<p>The aim of variation detection is to detect how many bases out of the total are different to a reference genome.</p>
<p>In Craig Venter’s genome 4.1 million DNA variants were reported.</p>
<p>What sort of variation could we find in the DNA sequencing?</p>
<ul>
<li>Single nucleotide variations (SNVs)</li>
<li>Single nucleotide polymorphisms (SNPs)</li>
<li>Small insertions and deletions (INDELs)</li>
<li>Large Chromosome rearrangements-Structural variations (SV)</li>
<li>Copy number variations (CNV)</li>
</ul>
<h3 id="variant-calling-vs-genotyping">Variant Calling vs genotyping<a class="headerlink" href="#variant-calling-vs-genotyping" title="Permanent link">&para;</a></h3>
<p>Variant calling is concerned with whether there is evidence of variant in a particular locus whereas genotyping talks about what the sets of alleles in that locus are and their frequencies. In haploid organisms variant calling and genotyping are equivalent whereas the same rule does not apply to other organisms.</p>
<p>Variant callers estimate the probability of a particular genotype given the observed data.</p>
<p>The question one would be asking is what possible genotypes would be possible for a sample. The remaining question is, given that our variant calling process calls a variant, does that mean that there is truly a variant in this locus and also given that the variant caller doesn’t detect a variant in a position does that mean there is no variant in that position.</p>
<p>The result of variant calling is a list of probable variants.</p>
<h3 id="process-of-variant-calling">Process of variant calling<a class="headerlink" href="#process-of-variant-calling" title="Permanent link">&para;</a></h3>
<p>Sample DNA -&gt; Sequencing -&gt; Read alignment -&gt; BAM file of aligned reads against reference genome -&gt; Genotyper -&gt; Variant list</p>
<p>The number of reads that stack up on each other is called <em>read coverage</em>. The data is converted into positional information of the reference with the read counts that have piled up under each position. Variant calling will look at how many bases out of the total number of bases is different to the reference at any position.</p>
<h3 id="homozygous-or-heterozygous-mutations">Homozygous or Heterozygous mutations:<a class="headerlink" href="#homozygous-or-heterozygous-mutations" title="Permanent link">&para;</a></h3>
<p>What should be noted about variants is that they are rare events and homozygous variants are even rarer than heterozygous events.</p>
<h2 id="variant-calling-software">Variant Calling Software:<a class="headerlink" href="#variant-calling-software" title="Permanent link">&para;</a></h2>
<p>There a number of software available for variant calling some of which are as follows:</p>
<ol>
<li>SAMtools (mpileup and bcftools): Li 2009 Bioinformatics</li>
<li>GATK: McKenna et al. 2010 Genome Res</li>
<li>FreeBayes: MIT</li>
<li>DiBayes: SOLiD software <a href="http://www.lifetechnologies.com">http://www.lifetechnologies.com</a></li>
<li>InGAP: Qi J, Zhao F, Buboltz A, Schuster SC.. 2009. <em>Bioinformatics</em></li>
<li>MAQGene: Bigelow H, Doitsidou M, Sarin S, Hobert O. 2009. <em>Nature Methods</em></li>
</ol>
<h3 id="variant-calling-using-samtools-mpileup-bcftools">Variant Calling using Samtools (Mpileup + bcftools)<a class="headerlink" href="#variant-calling-using-samtools-mpileup-bcftools" title="Permanent link">&para;</a></h3>
<p>Samtools calculates the genotype likelihoods. We then pipe the output to bcftools, which does our SNP calling based on those likelihoods.</p>
<p>Mpileup: Input: BAM file Output: Pileuped up reads under the reference</p>
<p>bcftools: Input: Pileup output from Mpileup Output: VCF file with sites and genotypes</p>
<p><a href="http://samtools.sourceforge.net/mpileup.shtml">Further information</a></p>
<h3 id="variant-calling-using-gatk-unified-genotyper">Variant Calling using GATK-Unified Genotyper<a class="headerlink" href="#variant-calling-using-gatk-unified-genotyper" title="Permanent link">&para;</a></h3>
<p>GATK is a programming framework based on the philosophy of MapReduce for developing NGS tools in a distributed or shared memory parallelized form.</p>
<p>GATK unified genotyper uses a Bayesian probabilistic model to calculate genotype likelihoods.</p>
<p>Inputs: BAM file</p>
<p>Output: VCF file with sites and genotypes.</p>
<p>The probability of a variant genotype for a given sequence of data is calculated using the <a href="http://en.wikipedia.org/wiki/Bayes%27_theorem">Bayes Theorem</a> as follows:</p>
<div class="highlight"><pre><span></span><code>P(Genotype | Data) =  (P(Data | Genotype) * P(Genotype)) / P(Data)
</code></pre></div>
<p>P(Genotype) is the overall probability of that genotype being present in a sequence. This is called the prior probability of a Genotype.</p>
<p>P(Data | Genotype) is the probability of the data (the reads) given the genotype</p>
<p>P(Data) is the probability of seeing the reads.</p>
<p>GATK unified genotyper is not very good in dealing with INDELs and thus we would only calculate SNPs throughout this tutorial. GATK is setup to work with diploid genomes but can be used on haploids as well.</p>
<p><a href="http://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_sting_gatk_walkers_genotyper_UnifiedGenotyper.html">Further information</a></p>
<h3 id="variant-calling-using-freebayes">Variant Calling using FreeBayes<a class="headerlink" href="#variant-calling-using-freebayes" title="Permanent link">&para;</a></h3>
<p>FreeBayes is a high performance, flexible variant caller which uses the open source Freebayes tool to detect genetic variations based high throughput sequencing data (BAM files).</p>
<p><a href="http://bioinformatics.bc.edu/marthlab/FreeBayes">Further information</a></p>
<h3 id="evaluation-of-detected-variants-using-variant-eval">Evaluation of detected variants using Variant Eval<a class="headerlink" href="#evaluation-of-detected-variants-using-variant-eval" title="Permanent link">&para;</a></h3>
<p>The identified variation can further be evaluated against known variations such as common dbSNPs. The result can be checked for high concordance to the common SNPs or a known set of SNPs, the truth set.</p>
<p>The results will have:</p>
<ul>
<li>True Positives (TP): The variants called by the software which are also a known variant in the known variants file.</li>
<li>False Positives (FP): The Variants called by the software which are not known to be variants in the known variants file.</li>
<li>True Negatives (TN): The variants not called by the software which are not known to be variants in the known variants file.</li>
<li>False Negatives (FN): The variants not called by the software which are known as variants in the known variants file.</li>
</ul>
<h3 id="quality-matrix">Quality Matrix:<a class="headerlink" href="#quality-matrix" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>TP | FP
---|----
TN | FN
</code></pre></div>
<p>Sensitivity: TP/(TP+FN)</p>
<p>Specificity: TN/(TN+FP)</p>
<p><em>Note:</em></p>
<p>Although software methods available can find variants in unique regions reliably, the short NGS read length prevent them from detecting variations in repetitive regions with comparable sensitivity.</p>
<p>DNA substitution mutations are of two types: Transitions and Transversions. The Ti/Tv ratio (Transitions/Transversions) is also an indicator of how well the model has performed for genotyping.</p>
<ul>
<li>
<p>Transition: a point mutation in which a purine nucleotide is changed to another purine nucleotide. (A\&lt;->G) or a pyrimidine nucleotide to another pyrimidine. Approximately 2 out of 3 SNPs are Transitions.</p>
</li>
<li>
<p>Transversion: a substitute of a purine for a pyrimidine.</p>
</li>
</ul>
<p>Although there are twice as many Transversions as there are Transitions because of the molecular mechanisms by which they are generated, Transition mutations occur at the higher rate than the Transversion mutations.</p>
<p>For more details on variant eval visit: <a href="http://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_sting_gatk_walkers_varianteval_VariantEval.html">http://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_sting_gatk_walkers_varianteval_VariantEval.html</a></p>
<p><strong>Notes:</strong></p>
<p>An important thing worth noting is the more data the better the variant calling. In addition multisampling improves performance.</p>
<h3 id="local-realignment">Local realignment<a class="headerlink" href="#local-realignment" title="Permanent link">&para;</a></h3>
<p>In order to call SNPs close by INDELs correctly, local realignment is strongly recommended before variant calling when using both UnifiedGenotyper and FreeBayes. Samtools mpileup output would not however be affected since it works around this by introducing Base Alignment Quality (BAC). For more information on BAC refer to: <a href="http://samtools.sourceforge.net/mpileup.shtml">http://samtools.sourceforge.net/mpileup.shtml</a></p>
<h2 id="the-galaxy-workflow-platform">The Galaxy workflow platform<a class="headerlink" href="#the-galaxy-workflow-platform" title="Permanent link">&para;</a></h2>
<p>Galaxy is an online bioinformatics workflow management system. Essentially, you upload your files, create various analysis pipelines and run them, then visualise your results.</p>
<p>Galaxy is really an interface to the various tools that do the data processing; each of these tools could be run from the command line, outside of Galaxy. Galaxy makes it easier to link up the tools together and visualise the entire analysis pipeline.</p>
<p>Galaxy uses the concept of &lsquo;histories&rsquo;. Histories are sets of data and workflows that act on that data. The data for this workshop is available in a shared history, which you can import into your own Galaxy account</p>
<p><a href="http://wiki.g2.bx.psu.edu/">Learn more about Galaxy here</a></p>
<p><strong>The Galaxy interface. Tools on the left, data in the middle, analysis workflow on the right.</strong></p>
<p><img alt="" src="../media/image03.png" /></p>
<h2 id="data-format-used-in-the-tutorial">Data Format used in the tutorial<a class="headerlink" href="#data-format-used-in-the-tutorial" title="Permanent link">&para;</a></h2>
<h4 id="sequence-alignment-map-format"><strong>Sequence Alignment Map format</strong><a class="headerlink" href="#sequence-alignment-map-format" title="Permanent link">&para;</a></h4>
<p><strong>SAM format</strong></p>
<p>Sequence Alignment/Map format records all information relevant to how a set of reads aligns to a reference genome. A SAM file has an optional set of header lines describing the context of the alignment, then one line per read, with the following format:</p>
<ul>
<li>11 mandatory fields (+ variable number of optional fields)</li>
<li>1 QNAME: Query name of the read</li>
<li>2 FLAG</li>
<li>3 RNAME: Reference sequence name</li>
<li>4 POS: Position of alignment in reference sequence</li>
<li>5 MAPQ: Mapping quality (Phred-scaled)</li>
<li>6 CIGAR: String that describes the specifics of the alignment against the reference</li>
<li>7 MRNM</li>
<li>8 MPOS</li>
<li>9 ISIZE</li>
<li>10 SEQQuery: Sequence on the same strand as the reference</li>
<li>11 QUAL: Query quality (ASCII-33=Phred base quality)</li>
</ul>
<p><strong>SAM example</strong></p>
<div class="highlight"><pre><span></span><code>    SRR017937.312 16 chr20 43108717 37 76M * 0 0
    TGAGCCTCCGGGCTATGTGTGCTCACTGACAGAAGACCTGGTCACCAAAGCCCGGGAAGAGCTGCAGGAAAAGCCG
    ?,@A=A\&lt;5=,@==A:BB@=B9(.;A@B;\&gt;@ABBB@@9BB@:@5\&lt;BBBB9)\&gt;BBB2\&lt;BBB@BBB?;;BABBBBBBB@
</code></pre></div>
<p>For this example:</p>
<ul>
<li><strong>QNAME = SRR017937.312</strong> - this is the name of this read</li>
<li><strong>FLAG = 16</strong> - see the format description below</li>
<li><strong>RNAME = chr20</strong> - this read aligns to chromosome 20</li>
<li><strong>POS = 43108717</strong> - this read aligns the sequence on chr20 at position 43108717</li>
<li><strong>MAPQ = 37</strong> - this is quite a high quality score for the alignment (b/w 0 and 90)</li>
<li><strong>CIGAR = 76M</strong> - this read aligns to the reference segment across all bases (76 Matches means no deletions or insertions. Note that &lsquo;aligns&rsquo; can mean &lsquo;aligns with mismatches&rsquo; - mismatches that don&rsquo;t affect the alignment are not recorded in this field)</li>
<li><strong>MRNM = *</strong> - see the format description below</li>
<li><strong>MPOS = 0</strong> as there is no mate for this read - the sequenced DNA library was single ended, not mate paired*.</li>
<li><strong>ISIZE = 0</strong> as there is no mate for this read</li>
<li><strong>SEQQuery =</strong> the 76bp sequence of the reference segment</li>
<li><strong>QUAL =</strong> per-base quality scores for each position on the alignment. This is just a copy of what is in the FASTQ file</li>
</ul>
<p><strong>SAM</strong> format is described more fully <a href="http://samtools.sourceforge.net/SAM1.pdf"><em>here</em></a></p>
<p>NOTE: reads are shown mapped to the &ldquo;sense&rdquo; strand of the reference, and bases are listed in 5&rsquo; -> 3&rsquo; order. This is important because an actual read might be from the other strand of DNA. The alignment tool will try to map the read as it is, and also the reverse compliment. If it was on the other strand then the reverse compliment is shown in the SAM file, rather than the original read itself</p>
<p>See<a href="http://www.illumina.com/technology/paired_end_sequencing_assay.ilmn">http://www.illumina.com/technology/paired_end_sequencing_assay.ilmn</a> for an overview of paired-end sequencing.</p>
<p><strong>SAM file in Galaxy</strong></p>
<p><img alt="" src="../media/image07.png" /></p>
<h4 id="binary-sequence-alignment-map-format">Binary Sequence Alignment Map format<a class="headerlink" href="#binary-sequence-alignment-map-format" title="Permanent link">&para;</a></h4>
<p><strong>BAM format</strong></p>
<p>SAM is a text format which is not space efficient. Binary Sequence Alignment is a compressed version of SAM.</p>
<p>Data in a BAM file is binary and therefore can&rsquo;t be visualised as text. If you try and visualise in Galaxy, it will default to downloading the file</p>
<p><strong>BAM file in IGV</strong></p>
<p><img alt="" src="../media/image01.png" /></p>
<h4 id="vcf-file-format">VCF file format<a class="headerlink" href="#vcf-file-format" title="Permanent link">&para;</a></h4>
<p><strong>What is VCF file:</strong></p>
<p>The <a href="http://vcftools.sourceforge.net/specs.html"><em>Variant Call Format</em></a> (VCF) is the emerging standard for storing variant data. Originally designed for SNPs and short INDELs, it also works for structural variations.</p>
<p>VCF consists of a header section and a data section. The header must contain a line starting with one &lsquo;#&lsquo;, showing the name of each field, and then the sample names starting at the 10<sup>th</sup> column. The data section is TAB delimited with each line consisting of at least 8 mandatory fields (the first 8 fields in the table below). The FORMAT field and sample information are allowed to be absent. We refer to the official <a href="http://vcftools.sourceforge.net/specs.html"><em>VCF spec</em></a> for a more rigorous description of the format.</p>
<table>
<thead>
<tr>
<th>Col</th>
<th>Field</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>CHROM</td>
<td>Chromosome name</td>
</tr>
<tr>
<td>2</td>
<td>POS</td>
<td>1-based position. For an indel, this is the position preceding the indel.</td>
</tr>
<tr>
<td>3</td>
<td>ID</td>
<td>Variant identifier. Usually the dbSNP rsID.</td>
</tr>
<tr>
<td>4</td>
<td>REF</td>
<td>Reference sequence at POS involved in the variant. For a SNP, it is a single base.</td>
</tr>
<tr>
<td>5</td>
<td>ALT</td>
<td>Comma delimited list of alternative sequence(s).</td>
</tr>
<tr>
<td>6</td>
<td>QUAL</td>
<td>Phred-scaled probability of all samples being homozygous reference.</td>
</tr>
<tr>
<td>7</td>
<td>FILTER</td>
<td>Semicolon delimited list of filters that the variant fails to pass.</td>
</tr>
<tr>
<td>8</td>
<td>INFO</td>
<td>Semicolon delimited list of variant information.</td>
</tr>
<tr>
<td>9</td>
<td>FORMAT</td>
<td>Colon delimited list of the format of individual genotypes in the following fields.</td>
</tr>
<tr>
<td>10+</td>
<td>Sample(s)</td>
<td>Individual genotype information defined by FORMAT.</td>
</tr>
</tbody>
</table>
<h4 id="vcf-format-in-galaxy">VCF format in Galaxy:<a class="headerlink" href="#vcf-format-in-galaxy" title="Permanent link">&para;</a></h4>
<p><img alt="" src="../media/image06.png" /></p>
<h4 id="bcf-file-format">Bcf file format:<a class="headerlink" href="#bcf-file-format" title="Permanent link">&para;</a></h4>
<h3 id="_1"><a class="headerlink" href="#_1" title="Permanent link">&para;</a></h3>
<p><strong>BCF format:</strong></p>
<p>BCF, or the binary variant call format, is the binary version of VCF. It keeps the same information in VCF, while much more efficient to process especially for many samples. The relationship between BCF and VCF is similar to that between BAM and SAM.</p>
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        <a href="../../bionitio/bionitio/" class="md-footer__link md-footer__link--prev" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Command-line software development
            </div>
          </div>
        </a>
      
      
        <a href="../../variant_calling_galaxy_1/variant_calling_galaxy_1/" class="md-footer__link md-footer__link--next" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Variant Calling part 1 (Galaxy)
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            <a href="https://www.melbournebioinformatics.org.au/">Melbourne Bioinformatics</a>
          </div>
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
        
      </div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../../..", "features": [], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing"}, "search": "../../../assets/javascripts/workers/search.fb4a9340.min.js", "version": null}</script>
    
    
      <script src="../../../assets/javascripts/bundle.a1c7c35e.min.js"></script>
      
    
  </body>
</html>